--- lib/irrlicht/source/Irrlicht/COpenGLExtensionHandler.h.orig	2013-12-17 21:29:50.324711941 +0100
+++ lib/irrlicht/source/Irrlicht/COpenGLExtensionHandler.h	2013-12-17 21:31:25.739819400 +0100
@@ -60,6 +60,10 @@
 	#endif
 	#include <GL/gl.h>
 	#include <GL/glx.h>
+	#include <GL/glxext.h>
+	#ifndef PFNGLBLENDEQUATIONPROC 
+         typedef void (APIENTRYP PFNGLBLENDEQUATIONPROC) (GLenum mode);
+        #endif
 	#if defined(_IRR_OPENGL_USE_EXTPOINTER_)
 	#include "glext.h"
 	#undef GLX_ARB_get_proc_address // avoid problems with local glxext.h
--- lib/irrlicht/source/Irrlicht/glxext.h.orig	2013-12-17 21:41:06.516874146 +0100
+++ lib/irrlicht/source/Irrlicht/glxext.h	2013-12-17 22:00:18.107664863 +0100
@@ -667,6 +667,8 @@
 #endif /* GLX_GLXEXT_PROTOTYPES */
 typedef Display * ( * PFNGLXGETCURRENTDISPLAYEXTPROC) (void);
 typedef int ( * PFNGLXQUERYCONTEXTINFOEXTPROC) (Display *dpy, GLXContext context, int attribute, int *value);
+
+typedef XID GLXContextID;
 typedef GLXContextID ( * PFNGLXGETCONTEXTIDEXTPROC) (const GLXContext context);
 typedef GLXContext ( * PFNGLXIMPORTCONTEXTEXTPROC) (Display *dpy, GLXContextID contextID);
 typedef void ( * PFNGLXFREECONTEXTEXTPROC) (Display *dpy, GLXContext context);
