diff -Naur a/configure b/configure
--- a/configure	2012-08-11 17:07:08.118260290 +0200
+++ b/configure	2012-08-11 17:06:40.968408145 +0200
@@ -4459,6 +4459,7 @@
 
   if test -z "${SYSTEM_JDK_DIR}"; then
     for dir in /usr/lib/jvm/java-openjdk /usr/lib/jvm/icedtea6 \
+    	      /usr/lib/jvm/java-7-openjdk /usr/lib/jvm/icedtea7 \
     	      /usr/lib/jvm/java-6-openjdk /usr/lib/jvm/openjdk \
               /usr/lib/jvm/java-icedtea /usr/lib/jvm/java-gcj /usr/lib/jvm/gcj-jdk \
               /usr/lib/jvm/cacao ; do
