diff -Naur icedtea-web-1.2-orig//plugin/icedteanp/IcedTeaJavaRequestProcessor.cc icedtea-web-1.2//plugin/icedteanp/IcedTeaJavaRequestProcessor.cc
--- icedtea-web-1.2-orig//plugin/icedteanp/IcedTeaJavaRequestProcessor.cc	2012-03-05 17:04:20.000000000 +0100
+++ icedtea-web-1.2//plugin/icedteanp/IcedTeaJavaRequestProcessor.cc	2012-05-17 15:23:23.963573382 +0200
@@ -37,6 +37,7 @@
 exception statement from your version. */
 
 #include <typeinfo>
+#include <unistd.h>
 
 #include "IcedTeaJavaRequestProcessor.h"
 #include "IcedTeaScriptablePluginObject.h"
diff -Naur icedtea-web-1.2-orig//plugin/icedteanp/IcedTeaPluginUtils.cc icedtea-web-1.2//plugin/icedteanp/IcedTeaPluginUtils.cc
--- icedtea-web-1.2-orig//plugin/icedteanp/IcedTeaPluginUtils.cc	2012-03-02 22:45:39.000000000 +0100
+++ icedtea-web-1.2//plugin/icedteanp/IcedTeaPluginUtils.cc	2012-05-17 15:17:45.742990605 +0200
@@ -40,6 +40,8 @@
 #include "IcedTeaScriptablePluginObject.h"
 #include "IcedTeaPluginUtils.h"
 
+#include <unistd.h>
+
 /**
  * Misc. utility functions used by the plugin
  */
