diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2012-10-09 00:30:11.534668663 +0000
+++ b/Makefile.in	2012-10-09 00:30:27.207914565 +0000
@@ -107,8 +107,8 @@
 GLIB2_V_216_LIBS = @GLIB2_V_216_LIBS@
 GLIB_CFLAGS = @GLIB_CFLAGS@
 GLIB_LIBS = @GLIB_LIBS@
-GTK_CFLAGS = @GTK_CFLAGS@
-GTK_LIBS = @GTK_LIBS@
+QT4_CFLAGS = @QT4_CFLAGS@
+QT4_LIBS = @QT4_LIBS@
 HG = @HG@
 ICEDTEA_REVISION = @ICEDTEA_REVISION@
 INSTALL = @INSTALL@
@@ -789,7 +789,7 @@
 @ENABLE_PLUGIN_TRUE@	  -DICEDTEA_WEB_JRE="\"$(SYSTEM_JRE_DIR)\"" \
 @ENABLE_PLUGIN_TRUE@	  -DPLUGIN_BOOTCLASSPATH=$(PLUGIN_BOOTCLASSPATH) \
 @ENABLE_PLUGIN_TRUE@	  $(GLIB_CFLAGS) \
-@ENABLE_PLUGIN_TRUE@	  $(GTK_CFLAGS) \
+@ENABLE_PLUGIN_TRUE@	  $(QT4_CFLAGS) \
 @ENABLE_PLUGIN_TRUE@	  $(MOZILLA_CFLAGS) \
 @ENABLE_PLUGIN_TRUE@	  -fPIC -o $@ -c $<
 
@@ -798,7 +798,7 @@
 @ENABLE_PLUGIN_TRUE@	$(CXX) $(CXXFLAGS) \
 @ENABLE_PLUGIN_TRUE@	  $(PLUGIN_OBJECTS) \
 @ENABLE_PLUGIN_TRUE@	  $(GLIB_LIBS) \
-@ENABLE_PLUGIN_TRUE@	  $(GTK_LIBS) \
+@ENABLE_PLUGIN_TRUE@	  $(QT4_LIBS) \
 @ENABLE_PLUGIN_TRUE@	  $(MOZILLA_LIBS)\
 @ENABLE_PLUGIN_TRUE@	  -shared -o $@
 
diff -Naur a/acinclude.m4 b/acinclude.m4
--- a/acinclude.m4	2012-10-09 00:30:11.521335339 +0000
+++ b/acinclude.m4	2012-10-09 00:43:56.296867625 +0000
@@ -359,34 +359,21 @@
 AC_MSG_RESULT(${enable_plugin})
 ])
 
-dnl ITW_GTK_CHECK_VERSION([gtk version])
-AC_DEFUN([ITW_GTK_CHECK_VERSION],
+dnl ITW_QT_CHECK_VERSION([])
+AC_DEFUN([ITW_QT_CHECK_VERSION],
 [
-  AC_MSG_CHECKING([for GTK$1 version])
-  GTK_VER=`$PKG_CONFIG --modversion gtk+-$1.0`
-  AC_MSG_RESULT([$GTK_VER])
+  AC_MSG_CHECKING([for Qt])
+  QT_VER=`$PKG_CONFIG --modversion QtGui`
+  AC_MSG_RESULT([$QT_VER])
 ])
 
-dnl ITW_GTK_CHECK([gtk version])
-AC_DEFUN([ITW_GTK_CHECK],
+dnl ITW_QT_CHECK([])
+AC_DEFUN([ITW_QT_CHECK],
 [
-  case "$1" in
-    default)
-      PKG_CHECK_MODULES(GTK, gtk+-3.0,
-        [ITW_GTK_CHECK_VERSION([3])],
-        [PKG_CHECK_MODULES(GTK, gtk+-2.0,
-           [ITW_GTK_CHECK_VERSION([2])],
-           [AC_MSG_ERROR([GTK $1 not found])]
-        )]
-      )
-      ;;
-    *)
-      PKG_CHECK_MODULES(GTK, gtk+-$1.0,
-        [ITW_GTK_CHECK_VERSION([$1])],
-        [AC_MSG_ERROR([GTK $1 not found])]
-      )
-      ;;
-  esac
+    PKG_CHECK_MODULES(QT, QtCore,
+        [ITW_QT_CHECK_VERSION([])],
+        [AC_MSG_ERROR([Qt not found])]
+    )
 ])
 
 AC_DEFUN_ONCE([IT_CHECK_PLUGIN_DEPENDENCIES],
@@ -395,20 +382,12 @@
 dnl FIXME: use unstable
 AC_REQUIRE([IT_CHECK_PLUGIN])
 if test "x${enable_plugin}" = "xyes" ; then
-  AC_ARG_WITH([gtk],
-    [AS_HELP_STRING([--with-gtk=[2|3|default]],
-    [the GTK+ version to use (default: 3)])],
-    [case "$with_gtk" in
-       2|3|default) ;;
-       *) AC_MSG_ERROR([invalid GTK version specified]) ;;
-     esac],
-    [with_gtk=default])
-  ITW_GTK_CHECK([$with_gtk])
+  ITW_QT_CHECK()
   PKG_CHECK_MODULES(GLIB, glib-2.0)
   AC_SUBST(GLIB_CFLAGS)
   AC_SUBST(GLIB_LIBS)
-  AC_SUBST(GTK_CFLAGS)
-  AC_SUBST(GTK_LIBS)
+  AC_SUBST(QT_CFLAGS)
+  AC_SUBST(QT_LIBS)
 
   PKG_CHECK_MODULES(MOZILLA, npapi-sdk, [
     AC_CACHE_CHECK([for xulrunner version], [xulrunner_cv_collapsed_version],[
diff -Naur a/plugin/icedteanp/IcedTeaNPPlugin.cc b/plugin/icedteanp/IcedTeaNPPlugin.cc
--- a/plugin/icedteanp/IcedTeaNPPlugin.cc	2012-10-09 00:30:11.564668490 +0000
+++ b/plugin/icedteanp/IcedTeaNPPlugin.cc	2012-10-09 00:30:27.211248205 +0000
@@ -145,7 +145,7 @@
 #define PLUGIN_FILE_EXTS "class,jar,zip"
 #define PLUGIN_MIME_COUNT 1
 
-#define FAILURE_MESSAGE "icedteanp plugin error: Failed to run %s." \
+#define FAILURE_MESSAGE "icedteanp plugin error: Failed to run %1." \
   "  For more detail rerun \"firefox -g\" in a terminal window."
 
 #if MOZILLA_VERSION_COLLAPSED < 1090100
@@ -1132,19 +1132,11 @@
 static void
 plugin_display_failure_dialog ()
 {
-  GtkWidget* dialog = NULL;
-
   PLUGIN_DEBUG ("plugin_display_failure_dialog\n");
 
-  dialog = gtk_message_dialog_new (NULL,
-                                   GTK_DIALOG_DESTROY_WITH_PARENT,
-                                   GTK_MESSAGE_ERROR,
-                                   GTK_BUTTONS_CLOSE,
-                                   FAILURE_MESSAGE,
-                                   appletviewer_executable);
-  gtk_widget_show_all (dialog);
-  gtk_dialog_run (GTK_DIALOG (dialog));
-  gtk_widget_destroy (dialog);
+  QMessageBox::critical(0, "Plugin display failure",
+                           QString(FAILURE_MESSAGE).arg(appletviewer_executable),
+                           QMessageBox::Ok);
 
   PLUGIN_DEBUG ("plugin_display_failure_dialog return\n");
 }
diff -Naur a/plugin/icedteanp/IcedTeaNPPlugin.h b/plugin/icedteanp/IcedTeaNPPlugin.h
--- a/plugin/icedteanp/IcedTeaNPPlugin.h	2012-10-09 00:30:11.564668490 +0000
+++ b/plugin/icedteanp/IcedTeaNPPlugin.h	2012-10-09 00:30:27.211248205 +0000
@@ -51,8 +51,8 @@
 #include <glib.h>
 #include <glib/gstdio.h>
 
-// GTK includes.
-#include <gtk/gtk.h>
+// QT4 includes.
+#include <QtGui/QMessageBox>
 
 #include "IcedTeaPluginUtils.h"
 #include "IcedTeaPluginRequestProcessor.h"
