--- kdeutils-4.5.0/printer-applet/printer-applet.py.orig	2010-07-07 19:13:14.000000000 +0200
+++ kdeutils-4.5.0/printer-applet/printer-applet.py		2010-07-29 23:34:32.000000000 +0200
@@ -237,10 +237,10 @@ class JobManager(QObject, monitor.Watche
             APPDIR = file.left(file.lastIndexOf('/'))
 
         self.mainWindow = MainWindow()
-        uic.loadUi(APPDIR + '/' + "printer-applet.ui", self.mainWindow)
+        uic.loadUi(unicode(APPDIR + '/' + "printer-applet.ui"), self.mainWindow)
 
         self.printersWindow = PrintersWindow(self)
-        uic.loadUi(APPDIR + '/' + "printer-applet-printers.ui", self.printersWindow)
+        uic.loadUi(unicode(APPDIR + '/' + "printer-applet-printers.ui"), self.printersWindow)
 
         self.sysTray = KStatusNotifierItem(self.mainWindow)
         self.sysTray.setCategory(KStatusNotifierItem.Hardware)
