diff -Naur a/data/stellarium.desktop b/data/stellarium.desktop
--- a/data/stellarium.desktop	2014-06-28 16:37:16.000000000 +0000
+++ b/data/stellarium.desktop	2014-09-20 07:31:17.604037042 +0000
@@ -5,6 +5,7 @@
 GenericName=Desktop planetarium
 GenericName[es]=Planetario de escritorio
 GenericName[ru]=Настольный планетарий
+GenericName[de]=Desktop-Planetarium
 Exec=stellarium
 Icon=stellarium
 StartupNotify=false
@@ -14,18 +15,5 @@
 Comment[es]=Planetario
 Comment[fr]=Planétarium
 Comment[ru]=Планетарий
+Comment[de]=Planetarium
 Keywords=astronomy;planetarium;
-
-Actions=Window;Fullscreen;Safemode;
-
-[Desktop Action Window]
-Name=Open in window
-Name[ru]=Открыть в окне
-Exec=stellarium -f no
-OnlyShowIn=Unity;
-
-[Desktop Action Fullscreen]
-Name=Open in fullscreen
-Name[ru]=Открыть в полноэкранном режиме
-Exec=stellarium -f yes
-OnlyShowIn=Unity;
