From 44d69b42159bcae721a7d24075bf2a39a5224f09 Mon Sep 17 00:00:00 2001
From: Diane Trout <diane@ghic.org>
Date: Wed, 15 Jan 2014 20:51:01 -0800
Subject: [PATCH 1/8] Require farstream 0.2

---
 libqtf/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libqtf/CMakeLists.txt b/libqtf/CMakeLists.txt
index 38d234b..1be5bb1 100644
--- a/libqtf/CMakeLists.txt
+++ b/libqtf/CMakeLists.txt
@@ -3,7 +3,7 @@
 # in their public headers instead of doing forward declarations.
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(TP_FARSTREAM REQUIRED telepathy-farstream)
-pkg_check_modules(FARSTREAM REQUIRED farstream-0.1)
+pkg_check_modules(FARSTREAM REQUIRED farstream-0.2)
 
 include_directories(
     ${CMAKE_CURRENT_BINARY_DIR}
-- 
1.8.3.1

