--- bitstream.h.orig	2016-01-20 16:26:24.889999999 +0000
+++ bitstream.h	2016-01-20 16:41:09.473333332 +0000
@@ -9,7 +9,7 @@
 void put1bit (Bit_stream_struc *, int);
 unsigned long look_ahead (Bit_stream_struc *, int);
 unsigned long getbits (Bit_stream_struc *, int);
-INLINE void putbits (Bit_stream_struc *, unsigned int, int);
+void putbits (Bit_stream_struc *, unsigned int, int);
 void byte_ali_putbits (Bit_stream_struc *, unsigned int, int);
 unsigned long byte_ali_getbits (Bit_stream_struc *, int);
 unsigned long sstell (Bit_stream_struc *);
