--- trunk/KDE/kdebindings/perl/qtdbus/tools/qdbusxml2perl/qdbusintrospection_p.h	2010/11/23 18:59:15	1200042
+++ trunk/KDE/kdebindings/perl/qtdbus/tools/qdbusxml2perl/qdbusintrospection_p.h	2010/11/23 19:00:56	1200043
@@ -63,7 +63,7 @@
 
 QT_BEGIN_NAMESPACE
 
-class Q_DBUS_EXPORT QDBusIntrospection
+class Q_DECL_EXPORT QDBusIntrospection
 {
 public:
     // forward declarations
-- trunk/KDE/kdebindings/perl/qtdbus/tools/qdbusxml2perl/qdbusmetaobject_p.h	2010/11/18 00:55:20	1198257
+++ trunk/KDE/kdebindings/perl/qtdbus/tools/qdbusxml2perl/qdbusmetaobject_p.h	2010/11/23 19:00:56	1200043
@@ -62,7 +62,7 @@
 class QDBusError;
 
 struct QDBusMetaObjectPrivate;
-struct Q_DBUS_EXPORT QDBusMetaObject: public QMetaObject
+struct Q_DECL_EXPORT QDBusMetaObject: public QMetaObject
 {
     bool cached;
 
