--- tomahawk-0.8.4.orig/src/libtomahawk/utils/TomahawkUtils.cpp	2015-04-15 05:41:40.000000000 +0100
+++ tomahawk-0.8.4/src/libtomahawk/utils/TomahawkUtils.cpp	2016-04-25 16:13:29.874059475 +0100
@@ -35,8 +35,8 @@
     #include <lastfm/ws.h>
 #endif
 
-#include <quazip/quazip.h>
-#include <quazip/quazipfile.h>
+#include <quazip5/quazip.h>
+#include <quazip5/quazipfile.h>
 
 // We need this for the version info (if available)
 #include <taglib/taglib.h>
