diff -Naur ClanLib-1.0.0/Sources/Display/Providers/png_provider_generic.cpp ClanLib-1.0.0-new/Sources/Display/Providers/png_provider_generic.cpp
--- ClanLib-1.0.0/Sources/Display/Providers/png_provider_generic.cpp	2006-10-27 03:55:01.000000000 +0200
+++ ClanLib-1.0.0-new/Sources/Display/Providers/png_provider_generic.cpp	2012-11-01 12:43:53.534732315 +0100
@@ -107,7 +107,11 @@
 		png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp) NULL);
 		cl_assert(false);
 	}   
-	if (setjmp(png_ptr->jmpbuf))
+#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
+   if (setjmp(png_jmpbuf(png_ptr)))
+#else
+   if (setjmp(png_ptr->jmpbuf))
+#endif
 	{
 		png_destroy_read_struct(&png_ptr, &info_ptr, &end_info);
 		cl_assert(false);
