--- startkde.cmake.orig	2009-01-15 14:24:44.000000000 +0100
+++ startkde.cmake	2009-01-15 14:33:08.000000000 +0100
 # Boot sequence:
 #
 # kdeinit is used to fork off processes which improves memory usage
@@ -206,7 +190,7 @@
 # For anything else (that doesn't set env vars, or that needs a window manager),
 # better use the Autostart folder.
 
-libpath=`kde4-config --path lib | tr : '\n'`
+libpath=`kde4-config --path lib | tr : '\n'`'\n/etc/kde/lib/'
 
 for prefix in `echo "$libpath" | sed -n -e 's,/lib[^/]*/,/env/,p'`; do
   for file in "$prefix"*.sh; do
