diff -Naur mainwindow_news.cpp mainwindow_news.cpp.new 
--- a/src/mainwindow_news.cpp 2014-04-21 01:46:15.979868438 +0200
+++ b/src/mainwindow_news.cpp 2014-04-21 01:53:13.234244353 +0200
@@ -40,7 +40,7 @@
 QString MainWindow::retrieveDistroNews(bool searchForLatestNews)
 {
   const QString ctn_ARCH_RSS = "https://www.archlinux.org/feeds/news/";
-  const QString ctn_CHAKRA_RSS = "http://chakra-project.org/news/index.php?/feeds/index.rss2";
+  const QString ctn_CHAKRA_RSS = "http://chakraos.org/news/index.php?/feeds/index.rss2";
   const QString ctn_MANJARO_RSS = "http://manjaro.org/feed/";
   
   LinuxDistro distro = UnixCommand::getLinuxDistro();
 
