--- a/src/backend/datasources/filters/FITSFilter.cpp	2017-04-02 15:47:32.000000000 +0800
+++ b/src/backend/datasources/filters/FITSFilter.cpp	2017-04-10 08:57:40.986675755 +0800
@@ -1268,7 +1268,7 @@
 
 		if (!updatedKeyword.comment.isEmpty()) {
 			if (fits_modify_comment(fitsFile, keywordUpdate.keyUpdated ? updatedKeyword.key.toLatin1() : originalKeyword.key.toLatin1(),
-			                        updatedKeyword.comment.toLatin1(), &status)) {
+			                        updatedKeyword.comment.toLatin1().data(), &status)) {
 				printError(status);
 				status = 0;
 			}
@@ -1336,7 +1336,7 @@
 
 	foreach (const FITSFilter::Keyword& keyword, keywords) {
 		if (keyword.updates.unitUpdated) {
-			if (fits_write_key_unit(fitsFile, keyword.key.toLatin1(), keyword.unit.toLatin1().constData(), &status)) {
+			if (fits_write_key_unit(fitsFile, keyword.key.toLatin1(), keyword.unit.toLatin1().data(), &status)) {
 				printError(status);
 				status = 0;
 			}
