diff --git a/src/lib/plugins/speeddial.cpp b/src/lib/plugins/speeddial.cpp
index 6072e8b..7e59e93 100644
--- a/src/lib/plugins/speeddial.cpp
+++ b/src/lib/plugins/speeddial.cpp
@@ -66,7 +66,11 @@ void SpeedDial::loadSettings()
         allPages = "url:\"https://www.qupzilla.com\"|title:\"QupZilla\";"
                    "url:\"http://blog.qupzilla.com\"|title:\"QupZilla Blog\";"
                    "url:\"https://github.com/QupZilla/qupzilla\"|title:\"QupZilla GitHub\";"
-                   "url:\"https://duckduckgo.com\"|title:\"DuckDuckGo\";";
+                   "url:\"https://chakralinux.org\"|title:\"The Chakra Project\";"
+                   "url:\"https://community.chakralinux.org\"|title:\"Forums\";"
+                   "url:\"https://webchat.freenode.net/?channels=chakra\"|title:\"IRC\";"
+                   "url:\"http://www.youtube.com/html5\"|title:\"YouTube-html5\";"
+                   "url:\"https://duckduckgo.com/?t=chakra\"|title:\"DuckDuckGo\";";
     }
     changed(allPages);
 
