From 58bba903cfc5fb05dd09b227b597c9487234bcaa Mon Sep 17 00:00:00 2001
From: Chaoting Liu <brli@chakraos.org>
Date: Fri, 30 Oct 2015 01:41:10 +0800
Subject: [PATCH] branded speeddial

---
 src/lib/plugins/speeddial.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/lib/plugins/speeddial.cpp b/src/lib/plugins/speeddial.cpp
index 07b791a..f003ff4 100644
--- a/src/lib/plugins/speeddial.cpp
+++ b/src/lib/plugins/speeddial.cpp
@@ -64,10 +64,11 @@ void SpeedDial::loadSettings()
     settings.endGroup();
 
     if (allPages.isEmpty()) {
-        allPages = "url:\"http://www.qupzilla.com\"|title:\"QupZilla\";"
-                   "url:\"http://blog.qupzilla.com\"|title:\"QupZilla Blog\";"
-                   "url:\"https://github.com/QupZilla/qupzilla\"|title:\"QupZilla GitHub\";"
-                   "url:\"https://duckduckgo.com\"|title:\"DuckDuckGo\";";
+        allPages = "url:\"https://chakraos.org\"|title:\"The Chakra Project\";"
+                   "url:\"https://chakraos.org/forum\"|title:\"Forums\";"
+                   "url:\"https://webchat.freenode.net/?channels=chakra\"|title:\"IRC\";"
+                   "url:\"http://www.youtube.com/html5\"|title:\"YouTube-html5\";"
+                   "url:\"https://duckduckgo.com/?t=chakra\"|title:\"DuckDuckGo\";";
     }
     changed(allPages);
 
-- 
2.4.1

