--- src/lib/preferences/preferences.cpp	2012-03-23 16:29:12.000000000 +0000
+++ src/lib/preferences/preferences.cpp	2012-03-23 13:42:51.095598055 +0000
@@ -94,7 +94,7 @@
     Settings settings;
     //GENERAL URLs
     settings.beginGroup("Web-URL-Settings");
-    m_homepage = settings.value("homepage", "qupzilla:start").toString();
+    m_homepage = settings.value("homepage", "http://www.chakra-linux.org/welcome.html").toString();
     m_newTabUrl = settings.value("newTabUrl", "qupzilla:speeddial").toString();
     ui->homepage->setText(m_homepage);
     ui->newTabUrl->setText(m_newTabUrl);
@@ -150,7 +150,7 @@
     //APPEREANCE
     m_themesManager = new ThemeManager(ui->themesWidget, this);
     settings.beginGroup("Browser-View-Settings");
-    ui->showStatusbar->setChecked(settings.value("showStatusBar", true).toBool());
+    ui->showStatusbar->setChecked(settings.value("showStatusBar", false).toBool());
     ui->showBookmarksToolbar->setChecked(p_QupZilla->bookmarksToolbar()->isVisible());
     ui->showNavigationToolbar->setChecked(p_QupZilla->navigationBar()->isVisible());
     ui->showHome->setChecked(settings.value("showHomeButton", true).toBool());
