--- src/app/qupzilla.cpp	2012-02-02 17:26:15.000000000 -0500
+++ src/app/qupzilla.cpp	2012-02-03 19:13:47.872107140 -0500
@@ -122,7 +122,7 @@
 
     Settings settings;
     settings.beginGroup("Web-URL-Settings");
-    int afterLaunch = settings.value("afterLaunch", 1).toInt();
+    int afterLaunch = settings.value("afterLaunch", 2).toInt();
     settings.endGroup();
     settings.beginGroup("SessionRestore");
     bool startingAfterCrash = settings.value("isCrashed", false).toBool();
@@ -531,7 +531,7 @@
 
     //Url settings
     settings.beginGroup("Web-URL-Settings");
-    m_homepage = settings.value("homepage", "qupzilla:start").toUrl();
+    m_homepage = settings.value("homepage", "qupzilla:speeddial").toUrl();
     m_newtab = settings.value("newTabUrl", "qupzilla:speeddial").toUrl();
     settings.endGroup();
 
@@ -546,7 +546,7 @@
     bool showBackForwardIcons = settings.value("showBackForwardButtons", true).toBool();
     bool showBookmarksToolbar = settings.value("showBookmarksToolbar", true).toBool();
     bool showNavigationToolbar = settings.value("showNavigationToolbar", true).toBool();
-    bool showMenuBar = settings.value("showMenubar", true).toBool();
+    bool showMenuBar = settings.value("showMenubar", false).toBool();
     bool showAddTab = settings.value("showAddTabButton", false).toBool();
     bool makeTransparent = settings.value("useTransparentBackground", false).toBool();
     m_sideBarWidth = settings.value("SideBarWidth", 250).toInt();
