--- src/lib/app/qupzilla.cpp.orig	2012-03-05 08:33:24.000000000 -0500
+++ src/lib/app/qupzilla.cpp	2012-03-05 14:33:05.027117810 -0500
@@ -138,7 +138,7 @@
 
     Settings settings;
     settings.beginGroup("Web-URL-Settings");
-    int afterLaunch = settings.value("afterLaunch", 1).toInt();
+    int afterLaunch = settings.value("afterLaunch", 2).toInt();
     settings.endGroup();
     settings.beginGroup("SessionRestore");
     bool startingAfterCrash = settings.value("isCrashed", false).toBool();
@@ -549,7 +549,7 @@
 
     //Url settings
     settings.beginGroup("Web-URL-Settings");
-    m_homepage = settings.value("homepage", "qupzilla:start").toUrl();
+    m_homepage = settings.value("homepage", "qupzilla:speeddial").toUrl();
     settings.endGroup();
 
     QWebSettings* websettings = mApp->webSettings();
@@ -563,7 +563,7 @@
     bool showBackForwardIcons = settings.value("showBackForwardButtons", true).toBool();
     bool showBookmarksToolbar = settings.value("showBookmarksToolbar", true).toBool();
     bool showNavigationToolbar = settings.value("showNavigationToolbar", true).toBool();
-    bool showMenuBar = settings.value("showMenubar", true).toBool();
+    bool showMenuBar = settings.value("showMenubar", false).toBool();
     bool showAddTab = settings.value("showAddTabButton", false).toBool();
     bool makeTransparent = settings.value("useTransparentBackground", false).toBool();
     m_sideBarWidth = settings.value("SideBarWidth", 250).toInt();
