diff --git a/src/lib/opensearch/searchenginesmanager.cpp b/src/lib/opensearch/searchenginesmanager.cpp
index 8f9281c..d09a2bc 100644
--- a/src/lib/opensearch/searchenginesmanager.cpp
+++ b/src/lib/opensearch/searchenginesmanager.cpp
@@ -167,7 +167,7 @@ void SearchEnginesManager::restoreDefaults()
     Engine duck;
     duck.name = "DuckDuckGo";
     duck.icon = QIcon(":/icons/sites/duck.png");
-    duck.url = "https://duckduckgo.com/?q=%s&t=qupzilla";
+    duck.url = "https://duckduckgo.com/?t=chakra&q=%s";
     duck.shortcut = "d";
 
     Engine sp;
diff --git a/src/lib/plugins/speeddial.cpp b/src/lib/plugins/speeddial.cpp
index 753cf66..fdc891f 100644
--- a/src/lib/plugins/speeddial.cpp
+++ b/src/lib/plugins/speeddial.cpp
@@ -47,17 +47,18 @@ void SpeedDial::loadSettings()
     Settings settings;
     settings.beginGroup("SpeedDial");
     QString allPages = settings.value("pages", QString()).toString();
-    m_backgroundImage = settings.value("background", QString()).toString();
-    m_backgroundImageSize = settings.value("backsize", "auto").toString();
+    m_backgroundImage = settings.value("file:///usr/share/icons/start-white.png", QString()).toString();
+    m_backgroundImageSize = settings.value("backsize", "contain").toString();
     m_maxPagesInRow = settings.value("pagesrow", 4).toInt();
     m_sizeOfSpeedDials = settings.value("sdsize", 231).toInt();
     m_sdcentered = settings.value("sdcenter", 0).toInt();
     settings.endGroup();
 
     if (allPages.isEmpty()) {
-        allPages = "url:\"http://www.qupzilla.com\"|title:\"QupZilla\";"
-                   "url:\"http://blog.qupzilla.com\"|title:\"QupZilla Blog\";"
-                   "url:\"https://github.com/QupZilla/qupzilla\"|title:\"QupZilla GitHub\";"
+        allPages = "url:\"http://chakraos.org\"|title:\"The Chakra Project - Welcome\";"
+                   "url:\"http://chakraos.org/forum\"|title:\"Forums\";"
+                   "url:\"http://webchat.freenode.net/?channels=chakra\"|title:\"IRC\";"
+                   "url:\"http://www.youtube.com/html5\"|title:\"YouTube-html5\";";
                    "url:\"https://duckduckgo.com\"|title:\"DuckDuckGo\";";
     }
     changed(allPages);
diff --git a/src/lib/preferences/preferences.cpp b/src/lib/preferences/preferences.cpp
index c431fb7..790727f 100644
--- a/src/lib/preferences/preferences.cpp
+++ b/src/lib/preferences/preferences.cpp
@@ -140,7 +140,7 @@ Preferences::Preferences(BrowserWindow* window, QWidget* parent)
     Settings settings;
     //GENERAL URLs
     settings.beginGroup("Web-URL-Settings");
-    m_homepage = settings.value("homepage", "qupzilla:start").toString();
+    m_homepage = settings.value("homepage", "http://chakraos.org/home/?welcome").toString();
     m_newTabUrl = settings.value("newTabUrl", "qupzilla:speeddial").toString();
     ui->homepage->setText(m_homepage);
     ui->newTabUrl->setText(m_newTabUrl);
@@ -215,7 +215,7 @@ Preferences::Preferences(BrowserWindow* window, QWidget* parent)
 
     //APPEREANCE
     settings.beginGroup("Browser-View-Settings");
-    ui->showStatusbar->setChecked(settings.value("showStatusBar", true).toBool());
+    ui->showStatusbar->setChecked(settings.value("showStatusBar", false).toBool());
     ui->showBookmarksToolbar->setChecked(settings.value("showBookmarksToolbar", true).toBool());
     ui->showNavigationToolbar->setChecked(settings.value("showNavigationToolbar", true).toBool());
     ui->showHome->setChecked(settings.value("showHomeButton", true).toBool());
