diff --git a/src/lib/opensearch/searchenginesmanager.cpp b/src/lib/opensearch/searchenginesmanager.cpp
index 8f9281c..d09a2bc 100644
--- a/src/lib/opensearch/searchenginesmanager.cpp
+++ b/src/lib/opensearch/searchenginesmanager.cpp
@@ -167,7 +167,7 @@ void SearchEnginesManager::restoreDefaults()
     Engine duck;
     duck.name = "DuckDuckGo";
     duck.icon = QIcon(":/icons/sites/duck.png");
-    duck.url = "https://duckduckgo.com/?q=%s&t=qupzilla";
+    duck.url = "https://duckduckgo.com/?t=chakra&q=%s";
     duck.shortcut = "d";
 
     Engine sp;