--- src/opensearch/searchenginesmanager.cpp	2012-02-10 08:24:13.000000000 -0500
+++ src/opensearch/searchenginesmanager.cpp	2012-02-10 08:28:37.871589581 -0500
@@ -43,7 +43,7 @@
 {
     Settings settings;
     settings.beginGroup("SearchEngines");
-    m_startingEngineName = settings.value("activeEngine", "Google").toString();
+    m_startingEngineName = settings.value("activeEngine", "DuckDuckGo").toString();
     settings.endGroup();
 
     connect(this, SIGNAL(enginesChanged()), this, SLOT(scheduleSave()));
@@ -128,7 +128,7 @@
     Engine duck;
     duck.name = "DuckDuckGo";
     duck.icon = QIcon(":/icons/menu/duck.png");
-    duck.url = "https://duckduckgo.com/?q=%s";
+    duck.url = "https://duckduckgo.com/?t=chakra&q=%s";
     duck.shortcut = "d";
 
     addEngine(google, false);
