--- src/opensearch/searchenginesmanager.cpp	2012-02-02 18:04:29.000000000 -0500
+++ src/opensearch/searchenginesmanager.cpp	2012-02-02 18:05:15.050820457 -0500
@@ -43,7 +43,7 @@
 {
     Settings settings;
     settings.beginGroup("SearchEngines");
-    m_startingEngineName = settings.value("activeEngine", "Google").toString();
+    m_startingEngineName = settings.value("activeEngine", "DuckDuckGo").toString();
     settings.endGroup();
 
     connect(this, SIGNAL(enginesChanged()), this, SLOT(scheduleSave()));
@@ -104,6 +104,12 @@
 
 void SearchEnginesManager::restoreDefaults()
 {
+    Engine duck;
+    duck.name = "DuckDuckGo";
+    duck.icon = QIcon(":/icons/menu/duck.png");
+    duck.url = "https://duckduckgo.com/?q=%s";
+    duck.shortcut = "d";
+    
     Engine google;
     google.name = "Google";
     google.icon = QIcon(":icons/menu/google.png");
@@ -125,12 +131,6 @@
     yt.shortcut = "yt";
     yt.suggestionsUrl = "http://suggestqueries.google.com/complete/search?ds=yt&output=firefox&q=%s";
 
-    Engine duck;
-    duck.name = "DuckDuckGo";
-    duck.icon = QIcon(":/icons/menu/duck.png");
-    duck.url = "https://duckduckgo.com/?q=%s";
-    duck.shortcut = "d";
-
     addEngine(google, false);
     addEngine(wiki, false);
     addEngine(yt, false);
