--- src/lib/opensearch/searchenginesmanager.cpp.orig	2012-07-11 12:37:18.000000000 -0400
+++ src/lib/opensearch/searchenginesmanager.cpp	2012-07-11 22:34:37.935484692 -0400
@@ -50,7 +50,7 @@
 {
     Settings settings;
     settings.beginGroup("SearchEngines");
-    m_startingEngineName = settings.value("activeEngine", "Google").toString();
+    m_startingEngineName = settings.value("activeEngine", "DuckDuckGo").toString();
     settings.endGroup();
 
     connect(this, SIGNAL(enginesChanged()), this, SLOT(scheduleSave()));
@@ -73,7 +73,7 @@
         // TODO: Remove when releasing new version
         //       and move into profile updater
         if (en.name == "DuckDuckGo") {
-            en.url = "https://duckduckgo.com/?q=%s&t=qupzilla";
+            en.url = "https://duckduckgo.com/?t=chakra&q=%s";
         }
 
         m_allEngines.append(en);
@@ -141,7 +141,7 @@
     Engine duck;
     duck.name = "DuckDuckGo";
     duck.icon = QIcon(":/icons/sites/duck.png");
-    duck.url = "https://duckduckgo.com/?q=%s&t=qupzilla";
+    duck.url = "https://duckduckgo.com/?t=chakra&q=%s";
     duck.shortcut = "d";
 
     addEngine(google);
