--- src/lib/plugins/speeddial.cpp	2012-02-10 07:50:20.000000000 -0500
+++ src/lib/plugins/speeddial.cpp	2012-02-10 07:53:53.317603600 -0500
@@ -38,18 +38,17 @@
     Settings settings;
     settings.beginGroup("SpeedDial");
     QString allPages = settings.value("pages", "").toString();
-    m_backgroundImage = settings.value("background", "").toString();
+    m_backgroundImage = settings.value("background", "file:///usr/share/icons/start-white.png").toString();
-    m_backgroundImageSize = settings.value("backsize", "auto").toString();
+    m_backgroundImageSize = settings.value("backsize", "fill").toString();
     m_maxPagesInRow = settings.value("pagesrow", 4).toInt();
-    m_sizeOfSpeedDials = settings.value("sdsize", 231).toInt();
+    m_sizeOfSpeedDials = settings.value("sdsize", 150).toInt();
     settings.endGroup();
 
     if (allPages.isEmpty()) {
-        allPages = "url:\"http://www.google.com\"|title:\"Google\";"
-                   "url:\"http://www.qupzilla.com\"|title:\"QupZilla\";"
-                   "url:\"http://blog.qupzilla.com\"|title:\"QupZilla Blog\";"
-                   "url:\"https://github.com/nowrep/QupZilla\"|title:\"QupZilla GitHub\";"
-                   "url:\"https://facebook.com\"|title:\"Facebook\";";
+        allPages = "url:\"http://www.chakra-linux.org\"|title:\"The Chakra Project - Welcome\";"
+                   "url:\"http://www.chakra-project.org/bbs\"|title:\"Forums\";"
+                   "url:\"http://webchat.freenode.net/?channels=chakra\"|title:\"IRC\";"
+                   "url:\"http://www.youtube.com/html5\"|title:\"YouTube-html5\";";
     }
     changed(allPages);
 
