diff --git a/src/lib/app/mainapplication.cpp b/src/lib/app/mainapplication.cpp
index 7bd4ec8..e3d3cac 100644
--- a/src/lib/app/mainapplication.cpp
+++ b/src/lib/app/mainapplication.cpp
@@ -130,7 +130,7 @@ MainApplication::MainApplication(int &argc, char** argv)
     DATADIR = qApp->applicationDirPath() + "/";
 #endif
 
-    setWindowIcon(QIcon(":icons/exeicons/qupzilla-window.png"));
+    setWindowIcon(QIcon::fromTheme("qupzilla", QIcon(":icons/exeicons/qupzilla-window.png")));
     bool noAddons = false;
     bool newInstance = false;
     QUrl startUrl;