--- src/strconstants.h.orig	2013-03-24 12:00:35.588965813 -0400
+++ src/strconstants.h	2013-03-24 12:48:33.828992616 -0400
@@ -43,15 +43,15 @@
   }
 
   static QString getForeignRepositoryName(){
-    return "AUR";
+    return "CCR";
   }
 
   static QString getArchLinuxNews(){
     return QObject::tr("Arch Linux news");
   }
 
-  static QString getManjaroLinuxNews(){
-    return QObject::tr("Manjaro Linux news");
+  static QString getChakraProjectNews(){
+    return QObject::tr("Chakra-Project news");
   }
 
   static QString getNewsErrorMessage(){
@@ -218,8 +218,8 @@
     return QObject::tr("Searching for Arch Linux latest news...");
   }
 
-  static QString getSearchingForManjaroLinuxNews(){
-    return QObject::tr("Searching for Manjaro Linux latest news...");
+  static QString getSearchingForChakraProjectNews(){
+    return QObject::tr("Searching for The Chakra Project latest news...");
   }
 
   static QString getOneOutdatedPackage(){
