--- src/unixcommand.h.orig	2013-03-23 22:06:08.000000000 -0400
+++ src/unixcommand.h	2013-03-24 12:25:45.989211848 -0400
@@ -33,7 +33,7 @@
 
 enum CommandExecuting { ectn_NONE, ectn_SYNC_DATABASE, ectn_SYSTEM_UPGRADE, ectn_INSTALL, ectn_REMOVE };
 
-enum LinuxDistro { ectn_ARCHLINUX, ectn_MANJAROLINUX, ectn_UNKNOWN };
+enum LinuxDistro { ectn_ARCHLINUX, ectn_CHAKRAPROJECT, ectn_UNKNOWN };
 
 //Forward class declarations.
 class QString;
