--- agent/AuthDialog.cpp	2010-04-10 10:13:18.634941750 +0200
+++ agent/AuthDialog.cpp	2010-04-10 10:13:56.021565836 +0200
@@ -169,7 +169,7 @@
         foreach(PolkitQt1::Identity *identity, identities) {
             // First check to see if the user is valid
             qDebug() << "User: " << identity;
-            KUser user = KUser::KUser(identity->toString().remove("unix-user:"));
+            KUser user = KUser(identity->toString().remove("unix-user:"));
             if (!user.isValid()) {
                 kWarning() << "User invalid: " << user.loginName();
                 continue;
