diff -Naur fontconfig-2.10.93-orig/conf.d/Makefile.am fontconfig-2.10.93/conf.d/Makefile.am
--- fontconfig-2.10.93-orig/conf.d/Makefile.am	2013-10-15 11:03:19.557640594 +0200
+++ fontconfig-2.10.93/conf.d/Makefile.am	2013-10-15 11:01:17.398166118 +0200
@@ -30,6 +30,7 @@
 	10-hinting.conf \
 	10-hinting-slight.conf \
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
diff -Naur fontconfig-2.10.93-orig/conf.d/Makefile.in fontconfig-2.10.93/conf.d/Makefile.in
--- fontconfig-2.10.93-orig/conf.d/Makefile.in	2013-10-15 11:03:19.557640594 +0200
+++ fontconfig-2.10.93/conf.d/Makefile.in	2013-10-15 11:02:37.945081593 +0200
@@ -304,6 +304,7 @@
 	10-hinting.conf \
 	10-hinting-slight.conf \
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
