diff -Naur trunk/common/config.py trunk.fixed/common/config.py
--- trunk/common/config.py	2013-10-14 14:06:32.104334000 +0000
+++ trunk.fixed/common/config.py	2013-10-14 14:17:04.978542330 +0000
@@ -1271,7 +1263,7 @@
             day = self.get_automatic_backup_day(profile_id)
             weekday = self.get_automatic_backup_weekday(profile_id)	
             period = str(self.get_automatic_backup_anacron_period(profile_id))
-            job_identify = profile_id + '_' + profile_name
+            job_identify = profile_id + '_' + profile_name.replace(' ', '_')
 
             if self.AT_EVERY_BOOT == backup_mode:
                 cron_line = 'echo "{msg}\n@reboot {cmd}"'
