--- src/kdegraphics-4.6.2/kruler/doc/CMakeLists.txt	2011-04-01 21:38:16.000000000 +0200
+++ Desenvolupament/CMakeLists.txt	2011-04-03 02:00:59.073215641 +0200
@@ -1,4 +1,4 @@
 ########### install files ###############
 #
 #
-kde4_create_handbook(index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/en)
+kde4_create_handbook(index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/en SUBDIR kruler)
\ No newline at end of file
