diff -rupN kdeconnect-kde-0.4.2/plasmoid/package/contents/ui/CompactRepresentation.qml kdeconnect-kde-0.4.2_new/plasmoid/package/contents/ui/CompactRepresentation.qml
--- kdeconnect-kde-0.4.2/plasmoid/package/contents/ui/CompactRepresentation.qml	2014-02-08 20:56:05.000000000 +0100
+++ kdeconnect-kde-0.4.2_new/plasmoid/package/contents/ui/CompactRepresentation.qml	2014-02-08 20:55:29.000000000 +0100
@@ -19,13 +19,20 @@
  */
 
 import QtQuick 1.1
-import org.kde.plasma.graphicswidgets 0.1 as PlasmaWidgets
 import org.kde.plasma.core 0.1 as PlasmaCore
 
 Item {
-    PlasmaWidgets.IconWidget {
+    id: view
+    anchors.fill: parent
+
+    //TODO: Use this to detect if we should be iconized or full size
+    function isConstrained() {
+        return (plasmoid.formFactor == Vertical || plasmoid.formFactor == Horizontal);
+    }
+
+    PlasmaCore.IconItem {
         id: icon
-        Component.onCompleted: setIcon("smartphone");
+        source: "smartphone"
         anchors.fill: parent
     }
 
@@ -41,4 +48,5 @@ Item {
             subText: "KDE Connect device notifications"
         }
     }
-}
\ Manca newline alla fine del file
+
+}
