--- Module.ui.orig	2015-04-23 12:00:21.000000000 +0200
+++ Module.ui	2015-04-30 23:43:52.205734776 +0200
@@ -11,17 +11,33 @@
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
-   <item row="6" column="1">
-    <widget class="QLabel" name="qt">
+   <item row="8" column="1">
+    <widget class="QLabel" name="label_3">
      <property name="text">
-      <string>Qt Version:</string>
+      <string>OS Type:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
-   <item row="3" column="3">
+   <item row="3" column="2">
+    <spacer name="verticalSpacer_2">
+     <property name="orientation">
+      <enum>Qt::Vertical</enum>
+     </property>
+     <property name="sizeType">
+      <enum>QSizePolicy::Fixed</enum>
+     </property>
+     <property name="sizeHint" stdset="0">
+      <size>
+       <width>20</width>
+       <height>24</height>
+      </size>
+     </property>
+    </spacer>
+   </item>
+   <item row="4" column="2">
     <widget class="QLabel" name="label_2">
      <property name="font">
       <font>
@@ -37,51 +53,77 @@
      </property>
     </widget>
    </item>
-   <item row="7" column="0">
-    <spacer name="horizontalSpacer_4">
+   <item row="9" column="2">
+    <spacer name="verticalSpacer_4">
      <property name="orientation">
-      <enum>Qt::Horizontal</enum>
+      <enum>Qt::Vertical</enum>
+     </property>
+     <property name="sizeType">
+      <enum>QSizePolicy::Fixed</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
-       <width>215</width>
-       <height>17</height>
+       <width>20</width>
+       <height>24</height>
       </size>
      </property>
     </spacer>
    </item>
-   <item row="6" column="3">
-    <widget class="QLabel" name="qtLabel">
+   <item row="12" column="2">
+    <widget class="QLabel" name="memoryLabel">
      <property name="text">
-      <string notr="true">{QtVersion}</string>
+      <string notr="true">{MemoryAmount}</string>
      </property>
     </widget>
    </item>
-   <item row="8" column="1">
-    <widget class="QLabel" name="label_3">
+   <item row="7" column="2">
+    <widget class="QLabel" name="kernelLabel">
      <property name="text">
-      <string>OS Type:</string>
+      <string notr="true">{KernelName}</string>
      </property>
-     <property name="alignment">
-      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
+    </widget>
+   </item>
+   <item row="1" column="0" colspan="4">
+    <widget class="QWidget" name="widget" native="true">
+     <layout class="QHBoxLayout" name="horizontalLayout" stretch="0">
+      <item>
+       <widget class="QLabel" name="logoLabel">
+        <property name="text">
+         <string notr="true">{LogoLabel}</string>
+        </property>
+        <property name="alignment">
+         <set>Qt::AlignCenter</set>
+        </property>
+       </widget>
+      </item>
+     </layout>
+    </widget>
+   </item>
+   <item row="6" column="2">
+    <widget class="QLabel" name="qtLabel">
+     <property name="text">
+      <string notr="true">{QtVersion}</string>
      </property>
     </widget>
    </item>
-   <item row="5" column="3">
-    <widget class="QLabel" name="plasmaLabel">
+   <item row="11" column="2">
+    <widget class="QLabel" name="processorLabel">
      <property name="text">
-      <string notr="true">{KDEVersion}</string>
+      <string notr="true">{PrcoessorName}</string>
      </property>
     </widget>
    </item>
-   <item row="8" column="3">
-    <widget class="QLabel" name="bitsLabel">
+   <item row="7" column="1">
+    <widget class="QLabel" name="kernel">
      <property name="text">
-      <string notr="true">{QtArchitecutre}</string>
+      <string>Kernel Version:</string>
+     </property>
+     <property name="alignment">
+      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
-   <item row="13" column="3">
+   <item row="13" column="2">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
@@ -94,82 +136,94 @@
      </property>
     </spacer>
    </item>
-   <item row="7" column="4">
-    <spacer name="horizontalSpacer_7">
-     <property name="orientation">
-      <enum>Qt::Horizontal</enum>
-     </property>
-     <property name="sizeHint" stdset="0">
-      <size>
-       <width>215</width>
-       <height>20</height>
-      </size>
-     </property>
-    </spacer>
-   </item>
-   <item row="11" column="3">
-    <widget class="QLabel" name="processorLabel">
-     <property name="text">
-      <string notr="true">{PrcoessorName}</string>
-     </property>
-    </widget>
-   </item>
-   <item row="12" column="1">
-    <widget class="QLabel" name="memory">
+   <item row="6" column="1">
+    <widget class="QLabel" name="qt">
      <property name="text">
-      <string>Memory:</string>
+      <string>Qt Version:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
-   <item row="11" column="1">
-    <widget class="QLabel" name="processor">
+   <item row="10" column="2">
+    <widget class="QLabel" name="label">
+     <property name="font">
+      <font>
+       <weight>75</weight>
+       <bold>true</bold>
+      </font>
+     </property>
      <property name="text">
-      <string notr="true">Processor:</string>
+      <string>Hardware</string>
      </property>
      <property name="alignment">
-      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
+      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
-   <item row="7" column="3">
-    <widget class="QLabel" name="kernelLabel">
+   <item row="8" column="2">
+    <widget class="QLabel" name="bitsLabel">
      <property name="text">
-      <string notr="true">{KernelName}</string>
+      <string notr="true">{QtArchitecutre}</string>
      </property>
     </widget>
    </item>
-   <item row="7" column="1">
-    <widget class="QLabel" name="kernel">
+   <item row="7" column="0">
+    <spacer name="horizontalSpacer_4">
+     <property name="orientation">
+      <enum>Qt::Horizontal</enum>
+     </property>
+     <property name="sizeHint" stdset="0">
+      <size>
+       <width>215</width>
+       <height>17</height>
+      </size>
+     </property>
+    </spacer>
+   </item>
+   <item row="11" column="1">
+    <widget class="QLabel" name="processor">
      <property name="text">
-      <string>Kernel Version:</string>
+      <string notr="true">Processor:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
-   <item row="12" column="3">
-    <widget class="QLabel" name="memoryLabel">
+   <item row="5" column="2">
+    <widget class="QLabel" name="plasmaLabel">
      <property name="text">
-      <string notr="true">{MemoryAmount}</string>
+      <string notr="true">{KDEVersion}</string>
      </property>
     </widget>
    </item>
-   <item row="5" column="1">
-    <widget class="QLabel" name="plasma">
+   <item row="7" column="3">
+    <spacer name="horizontalSpacer_7">
+     <property name="orientation">
+      <enum>Qt::Horizontal</enum>
+     </property>
+     <property name="sizeHint" stdset="0">
+      <size>
+       <width>215</width>
+       <height>20</height>
+      </size>
+     </property>
+    </spacer>
+   </item>
+   <item row="12" column="1">
+    <widget class="QLabel" name="memory">
      <property name="text">
-      <string>KDE Plasma Version:</string>
+      <string>Memory:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
-   <item row="2" column="3">
-    <spacer name="verticalSpacer_2">
+   <item row="0" column="2">
+    <spacer name="verticalSpacer_3">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
@@ -179,13 +233,29 @@
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
-       <height>24</height>
+       <height>60</height>
       </size>
      </property>
     </spacer>
    </item>
-   <item row="1" column="3" colspan="2">
+   <item row="5" column="1">
+    <widget class="QLabel" name="plasma">
+     <property name="text">
+      <string>KDE Plasma Version:</string>
+     </property>
+     <property name="alignment">
+      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
+     </property>
+    </widget>
+   </item>
+   <item row="2" column="0" colspan="4">
     <layout class="QVBoxLayout" name="verticalLayout">
+     <property name="sizeConstraint">
+      <enum>QLayout::SetDefaultConstraint</enum>
+     </property>
+     <property name="rightMargin">
+      <number>0</number>
+     </property>
      <item>
       <spacer name="verticalSpacer_6">
        <property name="orientation">
@@ -197,7 +267,7 @@
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
-         <height>40</height>
+         <height>20</height>
         </size>
        </property>
       </spacer>
@@ -207,6 +277,9 @@
        <property name="text">
         <string notr="true">{NameVersionLabel}</string>
        </property>
+       <property name="alignment">
+        <set>Qt::AlignCenter</set>
+       </property>
       </widget>
      </item>
      <item>
@@ -214,106 +287,16 @@
        <property name="text">
         <string notr="true">{UrlLabel}</string>
        </property>
+       <property name="alignment">
+        <set>Qt::AlignCenter</set>
+       </property>
        <property name="openExternalLinks">
         <bool>true</bool>
        </property>
       </widget>
      </item>
-     <item>
-      <spacer name="verticalSpacer_5">
-       <property name="orientation">
-        <enum>Qt::Vertical</enum>
-       </property>
-       <property name="sizeType">
-        <enum>QSizePolicy::Preferred</enum>
-       </property>
-       <property name="sizeHint" stdset="0">
-        <size>
-         <width>20</width>
-         <height>40</height>
-        </size>
-       </property>
-      </spacer>
-     </item>
     </layout>
    </item>
-   <item row="9" column="3">
-    <spacer name="verticalSpacer_4">
-     <property name="orientation">
-      <enum>Qt::Vertical</enum>
-     </property>
-     <property name="sizeType">
-      <enum>QSizePolicy::Fixed</enum>
-     </property>
-     <property name="sizeHint" stdset="0">
-      <size>
-       <width>20</width>
-       <height>24</height>
-      </size>
-     </property>
-    </spacer>
-   </item>
-   <item row="0" column="3">
-    <spacer name="verticalSpacer_3">
-     <property name="orientation">
-      <enum>Qt::Vertical</enum>
-     </property>
-     <property name="sizeType">
-      <enum>QSizePolicy::Maximum</enum>
-     </property>
-     <property name="sizeHint" stdset="0">
-      <size>
-       <width>20</width>
-       <height>24</height>
-      </size>
-     </property>
-    </spacer>
-   </item>
-   <item row="10" column="3">
-    <widget class="QLabel" name="label">
-     <property name="font">
-      <font>
-       <weight>75</weight>
-       <bold>true</bold>
-      </font>
-     </property>
-     <property name="text">
-      <string>Hardware</string>
-     </property>
-     <property name="alignment">
-      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
-     </property>
-    </widget>
-   </item>
-   <item row="1" column="1">
-    <widget class="QWidget" name="widget" native="true">
-     <layout class="QHBoxLayout" name="horizontalLayout">
-      <item>
-       <widget class="QLabel" name="logoLabel">
-        <property name="text">
-         <string notr="true">{LogoLabel}</string>
-        </property>
-       </widget>
-      </item>
-      <item>
-       <spacer name="horizontalSpacer">
-        <property name="orientation">
-         <enum>Qt::Horizontal</enum>
-        </property>
-        <property name="sizeType">
-         <enum>QSizePolicy::Fixed</enum>
-        </property>
-        <property name="sizeHint" stdset="0">
-         <size>
-          <width>4</width>
-          <height>0</height>
-         </size>
-        </property>
-       </spacer>
-      </item>
-     </layout>
-    </widget>
-   </item>
   </layout>
  </widget>
  <resources/>
