diff -Naur kuser-16.04.1/doc/CMakeLists.txt kuser-16.04.1-new/doc/CMakeLists.txt
--- kuser-16.04.1/doc/CMakeLists.txt	2015-11-15 01:32:18.000000000 +0800
+++ kuser-16.04.1-new/doc/CMakeLists.txt	2016-05-23 12:48:55.466732659 +0800
@@ -1,4 +1,4 @@
 ########### install files ###############
 #
 #
-kde4_create_handbook(index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/en SUBDIR kuser)
+KDOCTOOLS_CREATE_HANDBOOK(index.docbook INSTALL_DESTINATION ${HTML_INSTALL_DIR}/en SUBDIR kuser)

