--- a/Telegram/SourceFiles/base/base_pch.h	2018-12-31 10:53:16.232898792 +0100
+++ b/Telegram/SourceFiles/base/base_pch.h	2018-12-31 11:02:13.163096099 +0100
@@ -6,13 +6,13 @@
 https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 */
 
-#include <QtCore/QByteArray>
-#include <QtCore/QString>
-#include <QtCore/QUrl>
-#include <QtCore/QMutex>
-#include <QtCore/QRegularExpression>
-#include <QtCore/QThread>
-#include <QtCore/QCoreApplication>
+#include <qt5/QtCore/QByteArray>
+#include <qt5/QtCore/QString>
+#include <qt5/QtCore/QUrl>
+#include <qt5/QtCore/QMutex>
+#include <qt5/QtCore/QRegularExpression>
+#include <qt5/QtCore/QThread>
+#include <qt5/QtCore/QCoreApplication>
 
 #include <crl/crl.h>
 #include <rpl/rpl.h>

--- a/Telegram/SourceFiles/storage/storage_pch.h	2018-12-31 11:15:11.442604646 +0100
+++ b/Telegram/SourceFiles/storage/storage_pch.h_patched	2018-12-31 11:29:46.312487456 +0100
@@ -6,11 +6,11 @@
 https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 */
 
-#include <QtCore/QFile>
-#include <QtCore/QFileInfo>
-#include <QtCore/QDir>
-#include <QtCore/QByteArray>
-#include <QtCore/QString>
+#include <qt5/QtCore/QFile>
+#include <qt5/QtCore/QFileInfo>
+#include <qt5/QtCore/QDir>
+#include <qt5/QtCore/QByteArray>
+#include <qt5/QtCore/QString>
 
 #include <crl/crl.h>
 #include <rpl/rpl.h>

--- a/Telegram/SourceFiles/export/export_pch.h	2018-12-31 11:15:11.362607093 +0100
+++ b/Telegram/SourceFiles/export/export_pch.h_patched	2018-12-31 11:30:36.234288323 +0100
@@ -6,16 +6,16 @@
 https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 */
 
-#include <QtCore/QFile>
-#include <QtCore/QFileInfo>
-#include <QtCore/QDir>
-#include <QtCore/QSize>
-#include <QtCore/QTextStream>
-#include <QtCore/QDateTime>
-#include <QtCore/QString>
-#include <QtCore/QByteArray>
-#include <QtCore/QReadWriteLock>
-#include <QtCore/QRegularExpression>
+#include <qt5/QtCore/QFile>
+#include <qt5/QtCore/QFileInfo>
+#include <qt5/QtCore/QDir>
+#include <qt5/QtCore/QSize>
+#include <qt5/QtCore/QTextStream>
+#include <qt5/QtCore/QDateTime>
+#include <qt5/QtCore/QString>
+#include <qt5/QtCore/QByteArray>
+#include <qt5/QtCore/QReadWriteLock>
+#include <qt5/QtCore/QRegularExpression>
 
 #include <crl/crl.h>
 #include <rpl/rpl.h>
