--- a/Telegram/gyp/qt.gypi	2019-02-13 18:51:38.000000000 +0100
+++ b/Telegram/gyp/qt.gypi	2019-02-13 18:58:01.481532583 +0100
@@ -14,7 +14,7 @@
               [ 'build_macold', {
                 'qt_version%': '5.3.2',
               }, {
-                'qt_version%': '<!(echo /usr/include/qt/QtCore/*/ | grep -Po "\d+\.\d+\.\d+")',
+                'qt_version%': '<!(echo /usr/include/qt5/QtCore/*/ | grep -Po "\d+\.\d+\.\d+")',
               }]
             ],
           },
@@ -28,7 +28,7 @@
             [ 'build_macold', {
               'linux_path_qt%': '/usr/local/macold/Qt-<(qt_version)',
             }, {
-              'linux_path_qt%': '/usr/lib/qt',
+              'linux_path_qt%': '/usr/lib/qt5',
             }]
           ]
         },
@@ -153,13 +153,13 @@
 
   'include_dirs': [
     '/usr/include/qt',
-    '/usr/include/qt/QtCore',
-    '/usr/include/qt/QtGui',
-    '/usr/include/qt/QtDBus',
-    '/usr/include/qt/QtCore/<(qt_version)',
-    '/usr/include/qt/QtGui/<(qt_version)',
-    '/usr/include/qt/QtCore/<(qt_version)/QtCore',
-    '/usr/include/qt/QtGui/<(qt_version)/QtGui',
+    '/usr/include/qt5/QtCore',
+    '/usr/include/qt5/QtGui',
+    '/usr/include/qt5/QtDBus',
+    '/usr/include/qt5/QtCore/<(qt_version)',
+    '/usr/include/qt5/QtGui/<(qt_version)',
+    '/usr/include/qt5/QtCore/<(qt_version)/QtCore',
+    '/usr/include/qt5/QtGui/<(qt_version)/QtGui',
   ],
   'library_dirs': [
     '<(qt_loc)/lib',
--- a/Telegram/SourceFiles/base/base_pch.h	2018-12-31 10:53:16.232898792 +0100
+++ b/Telegram/SourceFiles/base/base_pch.h	2018-12-31 11:02:13.163096099 +0100
@@ -6,13 +6,13 @@
 https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 */
 
-#include <QtCore/QByteArray>
-#include <QtCore/QString>
-#include <QtCore/QUrl>
-#include <QtCore/QMutex>
-#include <QtCore/QRegularExpression>
-#include <QtCore/QThread>
-#include <QtCore/QCoreApplication>
+#include <qt5/QtCore/QByteArray>
+#include <qt5/QtCore/QString>
+#include <qt5/QtCore/QUrl>
+#include <qt5/QtCore/QMutex>
+#include <qt5/QtCore/QRegularExpression>
+#include <qt5/QtCore/QThread>
+#include <qt5/QtCore/QCoreApplication>
 
 #include <crl/crl.h>
 #include <rpl/rpl.h>

--- a/Telegram/SourceFiles/storage/storage_pch.h	2018-12-31 11:15:11.442604646 +0100
+++ b/Telegram/SourceFiles/storage/storage_pch.h_patched	2018-12-31 11:29:46.312487456 +0100
@@ -6,11 +6,11 @@
 https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 */
 
-#include <QtCore/QFile>
-#include <QtCore/QFileInfo>
-#include <QtCore/QDir>
-#include <QtCore/QByteArray>
-#include <QtCore/QString>
+#include <qt5/QtCore/QFile>
+#include <qt5/QtCore/QFileInfo>
+#include <qt5/QtCore/QDir>
+#include <qt5/QtCore/QByteArray>
+#include <qt5/QtCore/QString>
 
 #include <crl/crl.h>
 #include <rpl/rpl.h>

--- a/Telegram/SourceFiles/export/export_pch.h	2018-12-31 11:15:11.362607093 +0100
+++ b/Telegram/SourceFiles/export/export_pch.h_patched	2018-12-31 11:30:36.234288323 +0100
@@ -6,16 +6,16 @@
 https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 */
 
-#include <QtCore/QFile>
-#include <QtCore/QFileInfo>
-#include <QtCore/QDir>
-#include <QtCore/QSize>
-#include <QtCore/QTextStream>
-#include <QtCore/QDateTime>
-#include <QtCore/QString>
-#include <QtCore/QByteArray>
-#include <QtCore/QReadWriteLock>
-#include <QtCore/QRegularExpression>
+#include <qt5/QtCore/QFile>
+#include <qt5/QtCore/QFileInfo>
+#include <qt5/QtCore/QDir>
+#include <qt5/QtCore/QSize>
+#include <qt5/QtCore/QTextStream>
+#include <qt5/QtCore/QDateTime>
+#include <qt5/QtCore/QString>
+#include <qt5/QtCore/QByteArray>
+#include <qt5/QtCore/QReadWriteLock>
+#include <qt5/QtCore/QRegularExpression>
 
 #include <crl/crl.h>
 #include <rpl/rpl.h>

