--- src/happy-1.19.3/Setup.lhs	2014-04-23 09:18:45.632345133 +0000
+++ src/happy-1.19.3/Setup.lhs	2014-04-23 09:18:54.895460614 +0000
@@ -9,7 +9,7 @@
 import Distribution.Simple ( defaultMainWithHooks, simpleUserHooks, UserHooks(..) )
 import Distribution.Simple.LocalBuildInfo ( LocalBuildInfo(..) )
 import Distribution.Simple.Program
-import Distribution.Verbosity ( normal )
+
 
 import System.FilePath ((</>))
 import Control.Exception ( try )
