--- ConvertAll/source/convertall.py.old	2011-04-14 15:21:48.409861085 +0200
+++ ConvertAll/source/convertall.py	2011-04-14 15:22:56.159494054 +0200
@@ -151,8 +151,6 @@
     setupTranslator(app)  # must be before importing any convertall modules
     setLocalEncoding()
     import convertdlg
-    if not userStyle and not sys.platform.startswith('win'):
-        QtGui.QApplication.setStyle('plastique')
     win = convertdlg.ConvertDlg()
     win.show()
     signal.signal(signal.SIGINT, signal.SIG_IGN)
