diff --git a/data/postinstall.conf b/data/postinstall.conf
index 88f32c8..e32c681 100644
--- a/data/postinstall.conf
+++ b/data/postinstall.conf
@@ -2,10 +2,10 @@
 PKG_OVERLAY="/opt/chakra/pkgs"
 
 # PACKAGES TO BE REMOVED FROM THE LIVE SYSTEM
-REMOVE_LIVE_PACKAGES="chakra-hardware-detection tribe tribe-partitionmanager nmap kdeplasma-applets-welcome-plasmoid chd-nvidia chd-nvidia-173xx chd-nvidia-304xx chd-catalyst-legacy chd-catalyst chakra-init-live chakra-live-initcpio virtualbox-guest-additions"
+REMOVE_LIVE_PACKAGES="chakra-hardware-detection tribe nmap kdeplasma-applets-welcome-plasmoid chd-nvidia chd-nvidia-173xx chd-nvidia-304xx chd-nvidia-340xx chd-catalyst chakra-init-live chakra-live-initcpio virtualbox-guest-additions"
 
 # PACKAGES TO BE INSTALLED TO THE TARGET
-INSTALL_PACKAGES="partitionmanager octopi-notifier kapudan"
+INSTALL_PACKAGES="octopi-notifier kapudan"
 
 # KDE L10N PKG PREFIX
 KDE_L10N_PREFIX="kde-l10n"
diff --git a/scripts/postinstall-functions/20-job-install-grub2 b/scripts/postinstall-functions/20-job-install-grub2
index a0756e9..aa97d11 100644
--- a/scripts/postinstall-functions/20-job-install-grub2
+++ b/scripts/postinstall-functions/20-job-install-grub2
@@ -55,7 +55,7 @@ job_install_grub2()
     fi
 
     msg "generating []/boot/grub/grub.cfg"
-    chroot ${mountpoint} grub-mkconfig -o /boot/grub/grub.cfg --skip-resume >> /tmp/grub2.log 2>&1
+    chroot ${mountpoint} grub-mkconfig -o /boot/grub/grub.cfg >> /tmp/grub2.log 2>&1
 
     msg_job_done "job_install_grub2"
 }
