diff --git a/scripts/postinstall-functions/job-cleanup-drivers b/scripts/postinstall-functions/job-cleanup-drivers
index 751279ce0b77af9f35ccbea61105378e92babc3b..8ec848af34e6e2fa2072e2ac8e9dd55741a91e97 100644
--- a/scripts/postinstall-functions/job-cleanup-drivers
+++ b/scripts/postinstall-functions/job-cleanup-drivers
@@ -20,27 +20,20 @@ job_cleanup_drivers()
 	# remove any db.lck
 	chroot ${mountpoint} /bin/rm -f /var/lib/pacman/db.lck
 
-	USED_DRIVERS=`cat /etc/X11/xorg.conf | sed -n '/Section.*."Device"/,/EndSection/p' | grep -v "#" | grep Driver | cut -d '"' -f 2`
-	ALL_DRIVERS=`pacman -r ${mountpoint} -Q | grep xf86-video | cut -d "-" -f 3 | cut -d " " -f 1`
+	USED_MODULES=$(lsmod | cut -d' ' -f1)
+	ALL_DRIVERS=$(pacman -r ${mountpoint} -Q | grep xf86-video | cut -d "-" -f 3 | cut -d " " -f 1)
 
-	# replace found ati-drivers with ati and echo used drivers
 	touch /tmp/used_drivers
-	for driver in ${USED_DRIVERS}
+	[[ $(echo "$USED_MODULES" | grep "^radeon$") ]]  && echo "ati"     >> /tmp/used_drivers
+	[[ $(echo "$USED_MODULES" | grep "^i915$") ]]    && echo "intel"   >> /tmp/used_drivers
+
+	for driver in ${ALL_DRIVERS}
 	do
-		if [ "${driver}" = "radeon" ] ; then
-			echo "ati" >> /tmp/used_drivers
-		elif [ "${driver}" = "mach64" ] ; then
-			echo "ati" >> /tmp/used_drivers
-		elif [ "${driver}" = "r128" ] ; then
-			echo "ati" >> /tmp/used_drivers
-		else
-			echo "${driver}" >> /tmp/used_drivers
-		fi
+		[[ $(echo "$USED_MODULES" | grep "^${driver}$") ]] && echo "${driver}" >> /tmp/used_drivers
 	done
 
 	# reload real used drivers
-	unset ${USED_DRIVERS}
-	USED_DRIVERS=`cat /tmp/used_drivers`
+        USED_DRIVERS=$(cat /tmp/used_drivers)
 
 	# display found drivers
 	msg "configured driver: ${USED_DRIVERS}"
@@ -60,18 +53,24 @@ job_cleanup_drivers()
 	sed -i "/vesa/d" /tmp/remove_drivers
 
 	msg "cleanup drivers"
-	REMOVE_DRIVERS=`cat /tmp/remove_drivers`
-	for rdriver in ${REMOVE_DRIVERS}
-	do
-		chroot ${mountpoint} /usr/bin/pacman -Rn xf86-video-${rdriver} --noconfirm
-	done
-
-	msg "remove any unneeded dri pkgs"
-	REMOVE_DRI=`pacman -r ${mountpoint} -Qdtq | grep dri`
-	for rdri in ${REMOVE_DRI}
-	do
-		chroot ${mountpoint} /usr/bin/pacman -Rn ${rdri} --noconfirm
-	done
+        REMOVE_DRIVERS=$(cat /tmp/remove_drivers)
+	if [[ -n "$USED_DRIVERS" ]]; then
+		for rdriver in ${REMOVE_DRIVERS}
+		do
+			chroot ${mountpoint} /usr/bin/pacman -Rn xf86-video-${rdriver} --noconfirm
+		done
+		msg "remove any unneeded dri pkgs"
+		REMOVE_DRI=`pacman -r ${mountpoint} -Qdtq | grep dri`
+		for rdri in ${REMOVE_DRI}
+		do
+			chroot ${mountpoint} /usr/bin/pacman -Rn ${rdri} --noconfirm
+		done
+	else
+		msg "module not found > not removing any free drivers"
+		msg "output of lsmod:"
+		lsmod | sort
+		msg "output of lsmod done"
+	fi
 
 	msg "video driver removal complete"
 
diff --git a/scripts/postinstall-functions/job-configure-users b/scripts/postinstall-functions/job-configure-users
index fd590a14a93c8401ba5f88f149a311f18fd0c53d..97ffeb4a1cb912dd7a20227b1f6fefffa431ba79 100644
--- a/scripts/postinstall-functions/job-configure-users
+++ b/scripts/postinstall-functions/job-configure-users
@@ -37,6 +37,7 @@ job_configure_users()
 	chroot ${mountpoint} /bin/mkdir -p -v /home/${user_name}/.kde4/share/apps/konqueror
 	chroot ${mountpoint} /bin/mkdir -p -v /home/${user_name}/.local/share/applications
 	chroot ${mountpoint} /bin/mkdir -p -v /home/${user_name}/.kde4/share//kde4/services/searchproviders
+	chroot ${mountpoint} /bin/mkdir -p -v /home/${user_name}/.config/autostart
 
 	timestamp=`date +%F-%H%M%S`
 	chroot ${mountpoint} /bin/mkdir -p -v /home/${user_name}/.backup/${timestamp}
@@ -112,7 +113,7 @@ job_configure_users()
 	cp -vf /etc/skel/.bashrc ${mountpoint}/home/${user_name}/.bashrc
 	cp -vf /etc/skel/mimeapps.list ${mountpoint}/home/${user_name}/.local/share/applications/mimeapps.list
 	cp -vf /etc/skel/duckduckgo.desktop ${mountpoint}/home/${user_name}/.kde4/share/kde4/services/searchproviders/duckduckgo.desktop
-	cp -vf /etc/skel/kapudan.desktop ${mountpoint}/home/${user_name}/.kde4/share/kde4/Autostart/kapudan.desktop
+	cp -vf /etc/skel/kapudan.desktop ${mountpoint}/home/${user_name}/.config/autostart/kapudan.desktop
 	cp -vf /etc/skel/katerc ${mountpoint}/home/${user_name}/.kde4/share/config/katerc
 
 	chroot ${mountpoint} ln -sf /usr/bin/bluedevil-monolithic /home/${user_name}/.kde4/Autostart/bluedevil-monolithic
diff --git a/scripts/postinstall-functions/job-initialize-target b/scripts/postinstall-functions/job-initialize-target
index a703608dd2c9d7954af4a326a8bd48cbe589450f..f61828a1310096f168839dbc62a3fbd919bf5cf6 100644
--- a/scripts/postinstall-functions/job-initialize-target
+++ b/scripts/postinstall-functions/job-initialize-target
@@ -53,26 +53,6 @@ job_initialize_target()
 	sed -i -e 's~^.*#HaltCmd=.*~HaltCmd=/sbin/halt~' ${mountpoint}/usr/share/config/kdm/kdmrc
 	sed -i -e 's~^.*#RebootCmd=.*~RebootCmd=/sbin/reboot~' ${mountpoint}/usr/share/config/kdm/kdmrc
 
-	#msg "adjusting default wallpaper to screen resolution"
-	#echo "Search for your current screen resolution (this may flicker)"
-
-	#CURRENT_RES=`xrandr  | grep '*' | cut -b -16 | sed 's/ //g'`
-
-	#echo "Your current screen resolution is: $CURRENT_RES"
-	#echo "Try to link right backround ..."
-
-	#FOLDER="/usr/share/apps/kdm/themes/${THEME}/$CURRENT_RES"
-
-	#if [ -d $FOLDER ];
-	#then
-	#  echo "Using $FOLDER"
-	#cp -Rvf /usr/share/apps/kdm/themes/${THEME}/$CURRENT_RES/* ${mountpoint}/usr/share/apps/kdm/themes/${THEME}
-	#else
-	#  echo "Folder $FOLDER does not exists! Using 1280x800"
-	#  CURRENT_RES="1280x800"
-	#  cp -Rvf /usr/share/apps/kdm/themes/${THEME}/$CURRENT_RES/* ${mountpoint}/usr/share/apps/kdm/themes/${THEME}
-	#fi
-
 	msg "copy theme settings"
 	mkdir -p ${mountpoint}/root/.config
 	cp -v /home/live/.config/Trolltech.conf ${mountpoint}/root/.config/Trolltech.conf
diff --git a/scripts/postinstall.sh b/scripts/postinstall.sh
index e6d2cdaa543c9b63036a7ebffa6633343796bc51..dd17dda319b64d4b78e31329d6b12d917ba07646 100644
--- a/scripts/postinstall.sh
+++ b/scripts/postinstall.sh
@@ -67,9 +67,6 @@ fi
 # source the config
 . /usr/share/tribe/config/postinstall.conf
 
-# source nvidia-drv.conf
-. /etc/nvidia-drv.conf
-
 # load helper functions
 . /usr/share/tribe/scripts/helpers
 
