diff -ur kdenetwork-4.5.90/kget/transfer-plugins/bittorrent/bttransferfactory.cpp kdenetwork-4.5.90-libktorrent-stable/kget/transfer-plugins/bittorrent/bttransferfactory.cpp
--- kdenetwork-4.5.90/kget/transfer-plugins/bittorrent/bttransferfactory.cpp	2010-09-20 10:50:33.000000000 +0200
+++ kdenetwork-4.5.90-libktorrent-stable/kget/transfer-plugins/bittorrent/bttransferfactory.cpp	2010-11-22 21:46:11.000000000 +0100
@@ -18,18 +18,12 @@
 #include "advanceddetails/btadvanceddetailswidget.h"
 
 #include <kdebug.h>
-#include <util/functions.h>
 
 KGET_EXPORT_PLUGIN(BTTransferFactory)
 
 BTTransferFactory::BTTransferFactory(QObject *parent, const QVariantList &args)
   : TransferFactory(parent, args)
 {
-    if (!bt::InitLibKTorrent())
-    {
-        kError(5001) << "Failed to initialize libktorrent";
-        KGet::showNotification(0, "error", i18n("Cannot initialize libktorrent. Torrent support might not work."));
-    }
 }
 
 BTTransferFactory::~BTTransferFactory()
