--- kdenetwork-4.5.90-libktorrent-stable/kget/transfer-plugins/bittorrent/bttransferfactory.cpp.orig	2010-12-10 16:18:46.000000000 +0100
+++ kdenetwork-4.5.90-libktorrent-stable/kget/transfer-plugins/bittorrent/bttransferfactory.cpp	2010-12-24 23:38:25.853333340 +0100
@@ -19,7 +19,6 @@
 #include "advanceddetails/btadvanceddetailswidget.h"
 
 #include <kdebug.h>
-#include <util/functions.h>
 #include <version.h>
 #include <torrent/job.h>
 
@@ -28,11 +27,6 @@ KGET_EXPORT_PLUGIN(BTTransferFactory)
 BTTransferFactory::BTTransferFactory(QObject *parent, const QVariantList &args)
   : TransferFactory(parent, args)
 {
-    if (!bt::InitLibKTorrent())
-    {
-        kError(5001) << "Failed to initialize libktorrent";
-        KGet::showNotification(0, "error", i18n("Cannot initialize libktorrent. Torrent support might not work."));
-    }
 #if LIBKTORRENT_VERSION >= 0x010100
     bt::Job::setJobTracker(new BTJobTracker(this));
 #endif
