diff -up LibreCAD-2.2.0-rc1/librecad/src/ui/forms/qg_commandwidget.cpp.qt511 LibreCAD-2.2.0-rc1/librecad/src/ui/forms/qg_commandwidget.cpp
--- LibreCAD-2.2.0-rc1/librecad/src/ui/forms/qg_commandwidget.cpp.qt511	2018-07-23 13:44:03.713571700 -0400
+++ LibreCAD-2.2.0-rc1/librecad/src/ui/forms/qg_commandwidget.cpp	2018-07-23 13:44:16.145253360 -0400
@@ -27,6 +27,7 @@
 
 #include <algorithm>
 
+#include <QAction>
 #include <QKeyEvent>
 #include <QFileDialog>
 #include <QSettings>
diff -up LibreCAD-2.2.0-rc1/librecad/src/ui/generic/colorwizard.cpp.qt511 LibreCAD-2.2.0-rc1/librecad/src/ui/generic/colorwizard.cpp
--- LibreCAD-2.2.0-rc1/librecad/src/ui/generic/colorwizard.cpp.qt511	2018-07-23 13:44:27.832954070 -0400
+++ LibreCAD-2.2.0-rc1/librecad/src/ui/generic/colorwizard.cpp	2018-07-23 13:44:41.272609918 -0400
@@ -27,6 +27,7 @@
 #include "colorwizard.h"
 #include "ui_colorwizard.h"
 
+#include <QAction>
 #include <QColorDialog>
 #include <QLineEdit>
 #include <QListWidget>
diff -up LibreCAD-2.2.0-rc1/librecad/src/ui/generic/widgetcreator.cpp.qt511 LibreCAD-2.2.0-rc1/librecad/src/ui/generic/widgetcreator.cpp
--- LibreCAD-2.2.0-rc1/librecad/src/ui/generic/widgetcreator.cpp.qt511	2018-07-23 13:44:52.488322714 -0400
+++ LibreCAD-2.2.0-rc1/librecad/src/ui/generic/widgetcreator.cpp	2018-07-23 13:45:07.415940459 -0400
@@ -27,6 +27,8 @@
 #include "widgetcreator.h"
 #include "ui_widgetcreator.h"
 
+#include <QAction>
+#include <QActionGroup>
 #include <QSettings>
 #include <QLineEdit>
 #include <QPushButton>
