--- a/dropbox-scripts/dropbox_menu.sh	2012-01-26 17:36:05.000000000 +0100
+++ b/dropbox-scripts/dropbox_menu.sh	2015-02-28 13:19:32.172671524 +0100
@@ -34,7 +34,7 @@
 fi
 
 #Dropbox path detection
-dropbox_path=`${SCRIPTS_PATH}get_dropbox_folder.sh`
+dropbox_path="$HOME/Dropbox"
 #user_key=`sqlite3 ~/.dropbox/config.db 'SELECT value FROM config WHERE key="ns_p2p_key_map"' | awk 'NR==2 {print $0}' | sed 's/L//g'`
 echo $dropbox_path
 file_in_public="$dropbox_path"/Public/"${fileurl##*/}" #Path to file when it will be placed in dropbox public dir
