diff -Naur kdebase-runtime-4.3.90.orig/nepomuk/services/storage/nepomukstorage.notifyrc kdebase-runtime-4.3.90/nepomuk/services/storage/nepomukstorage.notifyrc
--- kdebase-runtime-4.3.90.orig/nepomuk/services/storage/nepomukstorage.notifyrc	2010-01-07 10:28:14.000000000 +0100
+++ kdebase-runtime-4.3.90/nepomuk/services/storage/nepomukstorage.notifyrc	2010-01-07 12:24:37.000000000 +0100
@@ -209,7 +209,7 @@
 Comment[x-test]=xxThe Nepomuk Semantic Desktop system could not be startedxx
 Comment[zh_CN]=无法启动 Nepomuk 语义桌面系统
 Comment[zh_TW]=Nepomuk 語意桌面系統無法啟動
-Action=Popup
+Action=
 
 [Event/convertingNepomukData]
 Name=Converting Nepomuk data
@@ -350,7 +350,7 @@
 Comment[x-test]=xxAll Nepomuk data is converted to a new storage backendxx
 Comment[zh_CN]=所有 Nepomuk 数据已经转换到新存储后端
 Comment[zh_TW]=Nepomuk 資料將被轉換到新的儲存後端介面。
-Action=Popup
+Action=
 
 [Event/convertingNepomukDataFailed]
 Name=Converting Nepomuk data failed
@@ -491,7 +491,7 @@
 Comment[x-test]=xxConverting Nepomuk data to a new backend failedxx
 Comment[zh_CN]=将 Nepomuk 数据转换到新后端的操作失败
 Comment[zh_TW]=轉換 Nepomuk 資料到新的後端介面時失敗。
-Action=Popup
+Action=
 
 [Event/convertingNepomukDataDone]
 Name=Converting Nepomuk data done
@@ -632,4 +632,4 @@
 Comment[x-test]=xxSuccessfully converted Nepomuk data to new backendxx
 Comment[zh_CN]=已成功将 Nepomuk 数据转换到新后端
 Comment[zh_TW]=已成功轉換 Nepomuk 資料到新的後端介面。
-Action=Popup
+Action=
