diff -ru minitube/src/qtsingleapplication/qtlocalpeer.cpp minitube.new/src/qtsingleapplication/qtlocalpeer.cpp
--- minitube/src/qtsingleapplication/qtlocalpeer.cpp	2012-03-03 14:42:25.000000000 +0000
+++ minitube.new/src/qtsingleapplication/qtlocalpeer.cpp	2012-06-03 21:45:05.983535234 +0000
@@ -58,6 +58,7 @@
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
