diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4db9ac5..251236d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,7 +43,15 @@ set_package_properties(KF5Attica PROPERTIES DESCRIPTION "A Qt library that imple
                        URL "https://projects.kde.org/attica"
                        TYPE OPTIONAL
                       )
+
+# Extra libraries
 set (HAVE_ATTICA ${KF5Attica_FOUND})
+if (HAVE_ATTICA)
+  set(XMLGUI_EXTRA_LIBS ${XMLGUI_EXTRA_LIBS} KF5::Attica)
+endif()
+if (WIN32)
+  set(XMLGUI_EXTRA_LIBS ${XMLGUI_EXTRA_LIBS} Segur32) # GetUserNameEx()
+endif()
 
 # Subdirectories
 add_subdirectory(src)
diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index 01725da..13b57fa 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -12,7 +12,7 @@ macro(XMLGUI_TESTS)
         add_executable(${_testname} ${_testname}.cpp)
         add_test(kxmlgui-${_testname} ${_testname})
         ecm_mark_as_test(${_testname})
-        target_link_libraries(${_testname} Qt5::Test KF5::I18n KF5::IconThemes KF5::XmlGui)
+        target_link_libraries(${_testname} Qt5::Test KF5::I18n KF5::IconThemes KF5::XmlGui ${XMLGUI_EXTRA_LIBS})
     endforeach(_testname)
 endmacro()
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a0dd642..e12a22e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,12 +1,5 @@
 add_subdirectory(ksendbugmail)
 
-if (HAVE_ATTICA)
-   set (XMLGUI_EXTRA_LIBS ${XMLGUI_EXTRA_LIBS} KF5::Attica)
-endif ()
-if(WIN32)
-    set (XMLGUI_EXTRA_LIBS ${XMLGUI_EXTRA_LIBS} Secur32) # GetUserNameEx()
-endif()
-
 configure_file(config-xmlgui.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-xmlgui.h )
 
 configure_file(
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 413fa92..c456ede 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -6,7 +6,7 @@ find_package(Qt5 5.2.0 CONFIG REQUIRED Test)
 macro(xmlgui_executable_tests)
   foreach(_testname ${ARGN})
     add_executable(${_testname} ${_testname}.cpp)
-    target_link_libraries(${_testname} Qt5::Test KF5::WidgetsAddons KF5::I18n KF5::XmlGui)
+    target_link_libraries(${_testname} Qt5::Test KF5::WidgetsAddons KF5::I18n KF5::XmlGui ${XMLGUI_EXTRA_LIBS})
     ecm_mark_as_test(${_testname})
   endforeach(_testname)
 endmacro()
diff --git a/tests/krichtexteditor/CMakeLists.txt b/tests/krichtexteditor/CMakeLists.txt
index 45c1abe..b8d8b56 100644
--- a/tests/krichtexteditor/CMakeLists.txt
+++ b/tests/krichtexteditor/CMakeLists.txt
@@ -9,7 +9,7 @@ include_directories(
    )
 add_executable(krichtexteditor ${krichtexteditorSources} )
 ecm_mark_as_test(krichtexteditor)
-target_link_libraries(krichtexteditor Qt5::Test KF5::ConfigWidgets KF5::TextWidgets KF5::XmlGui)
+target_link_libraries(krichtexteditor Qt5::Test KF5::ConfigWidgets KF5::TextWidgets KF5::XmlGui ${XMLGUI_EXTRA_LIBS})
 
 #install(TARGETS krichtexteditor DESTINATION ${BIN_INSTALL_DIR})
 #install(FILES krichtexteditorui.rc DESTINATION  ${DATA_INSTALL_DIR}/krichtexteditor)
