--- kio/kio/kprotocolmanager.cpp.orig   2010-11-05 11:48:26.000000000 +0100
+++ kio/kio/kprotocolmanager.cpp        2010-11-27 22:51:46.851725700 +0100
@@ -87,7 +87,7 @@ QL1S("Mozilla/5.0 (compatible; Konqueror
 QString::number(KDE::versionMajor()) % QL1C('.') % QString::number(KDE::versionMinor()) % \
 X % QL1S(") KHTML/") % \
 QString::number(KDE::versionMajor()) % QL1C('.') % QString::number(KDE::versionMinor()) % \
-QL1C('.') % QString::number(KDE::versionRelease()) % QL1S(" (like Gecko)")
+QL1C('.') % QString::number(KDE::versionRelease()) % QL1S(" (like Gecko) Chakra/1.0")
 
 #define PRIVATE_DATA \
 KProtocolManagerPrivate *d = kProtocolManagerPrivate