--- branches/KDE/4.5/kdelibs/kdoctools/customization/el/user.entities	2010/06/27 18:59:06	1143427
+++ branches/KDE/4.5/kdelibs/kdoctools/customization/el/user.entities	2010/12/01 17:35:40	1202632
@@ -6,3 +6,34 @@
      names (which if they occur will cost you a _lot_ of time)
  -->
 
+<!ENTITY Alt	"<keycap>Alt</keycap>">
+<!ENTITY Backspace	"<keycap>Backspace</keycap>">
+<!ENTITY cdrom	'<hardware>ÏƒÏ…ÏƒÎºÎµÏ…Î® CD-ROM</hardware>'>
+<!ENTITY Ctrl	"<keycap>Ctrl</keycap>">
+<!ENTITY dpi	'<acronym>dpi</acronym>'>
+<!ENTITY eg     "<abbrev>Ï€.Ï‡.</abbrev>">
+<!ENTITY Enter	"<keycap>Enter</keycap>">
+<!ENTITY Esc	"<keycap>Esc</keycap>">
+<!ENTITY etc    "<abbrev>ÎºÎ»Ï€.</abbrev>">
+<!ENTITY FAQ	'<acronym>FAQ</acronym>'>
+<!ENTITY ie     "<abbrev>i.e.</abbrev>">
+<!ENTITY infocenter	"<application>ÎšÎ­Î½Ï„ÏÎ¿ Ï€Î»Î·ÏÎ¿Ï†Î¿ÏÎ¹ÏŽÎ½</application>">
+<!ENTITY kcontrolcenter	"<application>ÎšÎ­Î½Ï„ÏÎ¿ ÎµÎ»Î­Î³Ï‡Î¿Ï… Ï„Î¿Ï… &kde;</application>">
+<!ENTITY kicon	"<guiicon>
+<inlinemediaobject>
+ <objectinfo><title>&kde;'s K-icon</title></objectinfo>
+ <imageobject><imagedata fileref='common/kmenu.png' format='PNG'/></imageobject>
+ <textobject><phrase>K</phrase></textobject>
+</inlinemediaobject>
+</guiicon> icon"><!-- if image is localised, then entityref should be used 
+                      instead of fileref -->
+<!ENTITY kmenu  "<guimenu>K</guimenu> menu">
+<!ENTITY LMB    "<mousebutton>Î±ÏÎ¹ÏƒÏ„ÎµÏÏŒ</mousebutton> ÎºÎ»Î¹Îº Ï€Î¿Î½Ï„Î¹ÎºÎ¹Î¿Ï">
+<!ENTITY MMB    "<mousebutton>Î¼ÎµÏƒÎ±Î¯Î¿</mousebutton> ÎºÎ»Î¹Îº Ï€Î¿Î½Ï„Î¹ÎºÎ¹Î¿Ï">
+<!ENTITY OS	"Î»ÎµÎ¹Ï„Î¿Ï…ÏÎ³Î¹ÎºÏŒ ÏƒÏÏƒÏ„Î·Î¼Î±"><!-- <acronym>OS</acronym>-->
+<!ENTITY partman "<application>Î”Î¹Î±Ï‡ÎµÎ¹ÏÎ¹ÏƒÏ„Î®Ï‚ ÎºÎ±Ï„Î±Ï„Î¼Î®ÏƒÎµÏ‰Î½ Ï„Î¿Ï… KDE</application>">
+<!ENTITY RMB    "<mousebutton>Î´ÎµÎ¾Î¯</mousebutton> ÎºÎ»Î¹Îº Ï€Î¿Î½Ï„Î¹ÎºÎ¹Î¿Ï">
+<!ENTITY Shift	"<keycap>Shift</keycap>">
+<!ENTITY systemsettings	"<application>Î¡Ï…Î¸Î¼Î¯ÏƒÎµÎ¹Ï‚ ÏƒÏ…ÏƒÏ„Î®Î¼Î±Ï„Î¿Ï‚</application>">
+<!ENTITY systemtray "<application>Ï€Î»Î±Î¯ÏƒÎ¹Î¿ ÏƒÏ…ÏƒÏ„Î®Î¼Î±Ï„Î¿Ï‚</application>">
+<!ENTITY Tab	"<keycap>Tab</keycap>">
--- branches/KDE/4.5/kdelibs/kdoctools/customization/el/user.entities	2010/12/01 17:35:40	1202632
+++ branches/KDE/4.5/kdelibs/kdoctools/customization/el/user.entities	2010/12/01 19:19:41	1202659
@@ -16,7 +16,7 @@
 <!ENTITY Esc	"<keycap>Esc</keycap>">
 <!ENTITY etc    "<abbrev>ÎºÎ»Ï€.</abbrev>">
 <!ENTITY FAQ	'<acronym>FAQ</acronym>'>
-<!ENTITY ie     "<abbrev>i.e.</abbrev>">
+<!ENTITY ie     "<abbrev>Î´Î·Î».</abbrev>">
 <!ENTITY infocenter	"<application>ÎšÎ­Î½Ï„ÏÎ¿ Ï€Î»Î·ÏÎ¿Ï†Î¿ÏÎ¹ÏŽÎ½</application>">
 <!ENTITY kcontrolcenter	"<application>ÎšÎ­Î½Ï„ÏÎ¿ ÎµÎ»Î­Î³Ï‡Î¿Ï… Ï„Î¿Ï… &kde;</application>">
 <!ENTITY kicon	"<guiicon>
