diff --git a/libs/handler.cpp b/libs/handler.cpp
index e247dc5..5232678 100644
--- a/libs/handler.cpp
+++ b/libs/handler.cpp
@@ -364,7 +364,7 @@ void Handler::editDialogAccepted()
 
 void Handler::replyFinished(QDBusPendingCallWatcher * watcher)
 {
-    QDBusPendingReply<QDBusObjectPath> reply = *watcher;
+    QDBusPendingReply<> reply = *watcher;
     if (reply.isError() || !reply.isValid()) {
         KNotification *notification = 0;
         QString error = reply.error().message();
