From c66a3097fa993fe68675c7b96a5e6a4d33e2a7f9 Mon Sep 17 00:00:00 2001
From: Fabian Kosmale <0inkane@googlemail.com>
Date: Tue, 29 Mar 2016 11:19:23 +0200
Subject: [PATCH] fix include

---
 model/kgpgsearchresultmodel.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/model/kgpgsearchresultmodel.h b/model/kgpgsearchresultmodel.h
index d754b12..bcacf57 100644
--- a/model/kgpgsearchresultmodel.h
+++ b/model/kgpgsearchresultmodel.h
@@ -22,9 +22,9 @@
 #include <QAbstractItemModel>
 
 #include <kgpgcompiler.h>
+#include <QStringList>
 
 class QString;
-class QStringList;
 
 class KGpgSearchResultModelPrivate;
 
-- 
2.7.4
