--- update-workspaces.sh.old	2010-09-10 22:15:49.000000000 +0200
+++ update-workspaces.sh	2010-12-12 12:38:39.000000000 +0100
@@ -42,4 +42,4 @@
 # If we're in bash then make HOSTTYPE available to Premake, for primitive arch-detection
 export HOSTTYPE="$HOSTTYPE"
 
-src/bin/premake --outpath ../workspaces/gcc --atlas --collada "$@" --target gnu || die "Premake failed"
+src/bin/premake --outpath ../workspaces/gcc --collada "$@" --target gnu || die "Premake failed"
