diff -Naur lilypond-2.12.3-orig/lily/spacing-spanner.cc lilypond-2.12.3/lily/spacing-spanner.cc
--- lilypond-2.12.3-orig/lily/spacing-spanner.cc	2010-01-12 16:20:49.000000000 +0000
+++ lilypond-2.12.3/lily/spacing-spanner.cc	2010-11-22 12:41:07.956666673 +0000
@@ -38,7 +38,7 @@
   vsize end = binary_search (all, (Grob*) me->get_bound (RIGHT),
 			     &Paper_column::less_than);  
   
-  all = vector<Grob*>::vector<Grob*> (all.begin () + start,
+  all = vector<Grob*> (all.begin () + start,
 				      all.begin () + end + 1);
   return all;
 }
