From 473676639868b5704fb6f1733bd48871a218f1bd Mon Sep 17 00:00:00 2001
From: Chaoting Liu <rainman59118@gmail.com>
Date: Fri, 9 Oct 2015 13:52:49 +0000
Subject: [PATCH 1/1] Add Chakra-specific search option

---
 Cutegram/cutegram.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cutegram/cutegram.cpp b/Cutegram/cutegram.cpp
index 9321800..2c8d7ef 100644
--- a/Cutegram/cutegram.cpp
+++ b/Cutegram/cutegram.cpp
@@ -179,9 +179,9 @@ Cutegram::Cutegram(QObject *parent) :
     p->translator = new QTranslator(this);
     p->theme = AsemanApplication::settings()->value("General/theme","Abrisham.qml").toString();
     p->emojisTheme = AsemanApplication::settings()->value("General/emojisTheme", "twitter").toString();
-    p->searchEngine = AsemanApplication::settings()->value("General/searchEngine","https://duckduckgo.com/?q=").toString();
+    p->searchEngine = AsemanApplication::settings()->value("General/searchEngine","https://duckduckgo.com/?t=chakra&q=").toString();
 
-    p->searchEngines = QStringList() << "https://duckduckgo.com/?q=" << "https://google.com/?q=" << "https://bing.com/?q=";
+    p->searchEngines = QStringList() << "https://duckduckgo.com/?t=chakra&q=" << "https://duckduckgo.com/?q=" << "https://google.com/?q=" << "https://bing.com/?q=";
 
 #ifdef Q_OS_ANDROID
     p->translationsPath = "assets:/files/translations";
-- 
2.4.1

