diff -Naur qwtplot3d/include/qwt3d_function.h qwtplot3d_new/include/qwt3d_function.h
--- qwtplot3d/include/qwt3d_function.h	2005-07-10 09:54:52.000000000 -0500
+++ qwtplot3d_new/include/qwt3d_function.h	2013-10-13 12:34:10.976112123 -0500
@@ -2,6 +2,7 @@
 #define qwt3d_function_h__2004_03_05_13_51_begin_guarded_code
 
 #include "qwt3d_gridmapping.h"
+#include <cstdio>
 
 namespace Qwt3D
 {
diff -Naur qwtplot3d/include/qwt3d_io_gl2ps.h qwtplot3d_new/include/qwt3d_io_gl2ps.h
--- qwtplot3d/include/qwt3d_io_gl2ps.h	2005-07-10 09:54:52.000000000 -0500
+++ qwtplot3d_new/include/qwt3d_io_gl2ps.h	2013-10-13 12:33:33.112940971 -0500
@@ -3,11 +3,7 @@
 
 #include <time.h>
 
-#if QT_VERSION < 0x040000
-#include <qgl.h>
-#else
-#include <QtOpenGL/qgl.h>
-#endif
+#include <QtOpenGL>
 
 #include "qwt3d_types.h"
 #include "qwt3d_io.h"
diff -Naur qwtplot3d/include/qwt3d_openglhelper.h qwtplot3d_new/include/qwt3d_openglhelper.h
--- qwtplot3d/include/qwt3d_openglhelper.h	2005-07-19 10:40:28.000000000 -0500
+++ qwtplot3d_new/include/qwt3d_openglhelper.h	2013-10-13 12:35:47.872363595 -0500
@@ -1,12 +1,8 @@
 #ifndef __openglhelper_2003_06_06_15_49__
 #define __openglhelper_2003_06_06_15_49__
 
-#include "qglobal.h"
-#if QT_VERSION < 0x040000
-#include <qgl.h>
-#else
-#include <QtOpenGL/qgl.h>
-#endif
+#include <QtOpenGL>
+#include <GL/glu.h>
 
 namespace Qwt3D
 {
