--- VirtualBox-5.1.16/configure.orig	2017-03-09 01:15:08.000000000 +0800
+++ VirtualBox-5.1.16/configure	2017-03-09 18:07:33.075191237 +0800
@@ -1694,8 +1694,7 @@
         if [ $? -eq 0 ]; then
           echo "(Qt5 from pkg-config)" >> $LOG
           FLGQT5=`pkg-config Qt5Core --cflags`
-          # gcc 4.8 is able to compile with C++11 (see also VBOX_GCC_std in Config.kmk)
-          [ $cc_maj -eq 4 -a $cc_min -eq 8 ] && FLGQT5="$FLGQT5 -std=c++11"
+          FLGQT5="$FLGQT5 -std=c++11"
           INCQT5=`strip_I "$FLGQT5"`
           LIBDIR5=`pkg-config Qt5Core --variable=libdir`
           LIBQT5=`pkg-config Qt5Core --libs`
