--- VirtualBox-5.1.2/configure.orig	2016-07-21 23:15:11.118121629 +0100
+++ VirtualBox-5.1.2/configure	2016-07-21 23:17:07.133529084 +0100
@@ -1691,6 +1691,7 @@
         if [ $? -eq 0 ]; then
           echo "(Qt5 from pkg-config)" >> $LOG
           FLGQT5=`pkg-config Qt5Core --cflags`
+          FLGQT5="-std=gnu++11 $FLGQT5"
           INCQT5=`strip_I "$FLGQT5"`
           LIBDIR5=`pkg-config Qt5Core --variable=libdir`
           LIBQT5=`pkg-config Qt5Core --libs`
