--- src/backends/x11.cmake      2014-05-12 22:49:39.000000000 +0000
+++ src/backends/x11.cmake.new  2014-05-30 00:31:48.135916437 +0000
@@ -8,8 +8,8 @@
     message(FATAL_ERROR "Xinput not found")
 endif()
 
-pkg_search_module(SYNAPTICS REQUIRED xorg-synaptics)
-pkg_search_module(XORG REQUIRED xorg-server)
+#pkg_search_module(SYNAPTICS REQUIRED xorg-synaptics)
+#pkg_search_module(XORG REQUIRED xorg-server)
 
 include_directories(${XCB_INCLUDE_DIR}
                     ${X11_XCB_INCLUDE_DIR}

