diff --git a/src/packageui/bundlemodel.cpp b/src/packageui/bundlemodel.cpp
index d43dd8f..65bc2b7 100644
--- a/src/packageui/bundlemodel.cpp
+++ b/src/packageui/bundlemodel.cpp
@@ -28,6 +28,8 @@
 #include "utils.h"
 
 //KDE Includes
+#include <KXmlGuiWindow>
+#include <KActionCollection>
 #include <KDesktopFile>
 #include <KIO/Job>
 #include <KIO/FileCopyJob>
@@ -35,6 +37,7 @@
 #include <KUrl>
 
 //Qt Includes
+#include <QAction>
 #include <QProcess>
 #include <QtConcurrentRun>
 #include <QFuture>
@@ -42,7 +45,7 @@
 //BundleModel
 BundleModel::BundleModel(QObject* parent)
     : QSortFilterProxyModel(parent)
-    , m_sourceModel(new BundleModelPrivate(this))
+    , m_sourceModel(new BundleModelPrivate(parent))
     , m_showInstalledOnly(false)
 {
     setSourceModel(m_sourceModel);
@@ -90,6 +93,8 @@ BundleModelPrivate::BundleModelPrivate(QObject* parent)
     , m_currentArch(Utils::getArchitecture())
     , m_process(new QProcess(this))
 {
+    m_xmlGui = qobject_cast<KXmlGuiWindow*>(parent);
+
     //needed to use the model with qml
     QHash<int, QByteArray> rNames;
     rNames.insert(BundleRoles::Name, "name");
@@ -269,6 +274,7 @@ void BundleModelPrivate::runBundle(const QModelIndex index)
 
 void BundleModelPrivate::downloadBundle(const int& index)
 {
+    m_xmlGui->actionCollection()->action("reload")->setEnabled(false);
     QString bundleName = m_info[index].completeName + "-" + m_currentArch;
     KUrl url ("http://chakra-project.org/repo/bundles/" + m_currentArch + "/" + bundleName + ".cb");
 
@@ -320,6 +326,7 @@ void BundleModelPrivate::setFinishedPercentage(KJob* job, long unsigned int perc
 
 void BundleModelPrivate::jobFinished(KJob* job)
 {
+    m_xmlGui->actionCollection()->action("reload")->setEnabled(true);
     const int index = job->property("index").toInt();
 
     //aborted jobs
diff --git a/src/packageui/bundlemodel.h b/src/packageui/bundlemodel.h
index d9d9141..c8ae652 100644
--- a/src/packageui/bundlemodel.h
+++ b/src/packageui/bundlemodel.h
@@ -24,6 +24,9 @@
 //Project Includes
 #include "defines.h"
 
+//KDE Includes
+#include <KXmlGuiWindow>
+
 //Qt Includes
 #include <QAbstractItemModel>
 #include <QSortFilterProxyModel>
@@ -129,7 +132,9 @@ private slots:
     void setFinishedPercentage(KJob *job, unsigned long percent);
     void jobFinished(KJob *job);
     void setSpeed(KJob *job, ulong speed);
+
 private:
+    KXmlGuiWindow *m_xmlGui;
     QList<BundleInfo> m_info;
     BundleInfo infoForInstalledBundle(const QString &BundlePath) const;
     QHash<int, bool> m_runningJobs;
