Author: C. Boemann <cbo@boemann.dk>
Date:   Mon Jul 29 11:45:49 2013 +0200

    Fix crash
    
    BUG: 319155

diff --git a/libs/textlayout/KoTextLayoutArea_paint.cpp b/libs/textlayout/KoTextLayoutArea_paint.cpp
index e94cc71..c633913 100644
--- a/libs/textlayout/KoTextLayoutArea_paint.cpp
+++ b/libs/textlayout/KoTextLayoutArea_paint.cpp
@@ -839,7 +839,7 @@ void KoTextLayoutArea::decorateParagraph(QPainter *painter, QTextBlock &block, b
 
     QList<KoTextBlockData::MarkupRange>::Iterator markIt = blockData.markupsBegin(KoTextBlockData::Misspell);
     QList<KoTextBlockData::MarkupRange>::Iterator markEnd = blockData.markupsEnd(KoTextBlockData::Misspell);
-    for (int i = 0 ; i <= layout->lineCount(); ++i) {
+    for (int i = 0 ; i < layout->lineCount(); ++i) {
         if (markIt == markEnd) {
             break;
         }
